---
title: "Comparing all data"
author: "João Santos"
date: "2025-06-02"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data Preparation

In this step we will merge all the data from different datasets. The first one that was used had 2 replicates per condition with one assigned to the Proliferative state and the other one to the Senescence condition. Here they used BJ cells. All this data is RNA-seq data and this one in specific was only Poly-A RNAs

This dataset was named data1, and the samples were as following:
SRR26383169 --> Senescence
SRR26383170 --> Senescence
SRR26383171 --> Young
SRR26383172 --> Young

The next dataset had 2 replicates per condition, they used foreskin fibroblasts (BJ) but now with more conditions comparing with data1. Here they also quantified only Poly-A containing RNAs

These are the samples:
SRR9601022 --> Proliferative
SRR9601023 --> Proliferative
SRR9601024 --> Quiescent
SRR9601025 --> Quiescent
SRR9601026 --> Senescence
SRR9601027 --> Senescence
SRR9601028 --> Deep Senescence
SRR9601029 --> Deep Senescence

Since we can only use 2 conditions to compare each time, we did the following comparisons/subsets:
Proliferative vs Deep --> data2
Proliferative vs Quiescent --> data3
Proliferative vs Senescence --> data4
Senescence vs Deep Senescence --> data5


The next dataset had 2 replicates per condition, one for Proliferative and another for Senescence, however in this experiment they used ribodeplition to sequence total RNA.

This dataset is named data6 and these are the samples and conditions:
SRR9016157 --> Proliferative
SRR9016158 --> Proliferative
SRR9016161 --> Senescence
SRR9016162 --> Senescence


The next dataset has 3 replicates per condition , and uses a different cell line called "MDAH041_LLP". There are 3 conditions, one young (PD10-12), ageing (PD17-19) and replicatively senescence (PD28-30). 

The samples are as follows:
SRR1544483 --> Proliferative 
SRR1544484 --> Proliferative
SRR1544485 --> Proliferative 
SRR1544486 --> Quiescent ??
SRR1544487 --> Quiescent ?? 
SRR1544488 --> Quiescent ??
SRR1544489 --> Senescence
SRR1544490 --> Senescence
SRR1544491 --> Senescence




# Data1

This data was already collected in the Markdown named VAST_analysis_RM.Rmd 

```{r}
library(readr)
library(UpSetR)
library(dplyr)
library(tidyr)
library(patchwork)
library(ggrepel)
library(viridis) 
library(ggplot2)
library(patchwork)
library(ggpubr)
library(Cairo)
library(tidyverse)
library(clusterProfiler)
library(org.Hs.eg.db)

# This is the object to be used for data1

data1= data
```

```{r}
selected_events <- c("S", "IR", "Alt3", "Alt5")

# Prepare the data for plotting by filtering and summarizing
plot_data_filtered <- output_transform_all %>%
  # *** THE KEY FILTERING STEP IS HERE ***
  filter(COMPLEXAS %in% selected_events) %>%
  
  # Now, proceed with counting and arranging as before
  count(COMPLEXAS, name = "count") %>%
  mutate(
    percentage = (count / sum(count)) * 100,
    # Reorder the 'COMPLEX' factor based on frequency for a clean plot
    COMPLEXAS = fct_reorder(COMPLEXAS, count, .desc = TRUE) 
  )

# Let's inspect the filtered and prepared data (optional)
print(plot_data_filtered)


# --- 3. PLOTTING: Create the publication-quality bar chart for the selected events ---
all_events=ggplot(plot_data_filtered, aes(x = COMPLEXAS, y = count, fill = COMPLEXAS)) +
  
  # Create the bars
  geom_bar(stat = "identity", color = "black", alpha = 0.8) +
  
  # Add data labels on top of each bar (count and percentage of the subset)
  geom_text(aes(label = paste0(count, "\n(", round(percentage, 1), "%)")), 
            vjust = -0.5, # Position the text slightly above the bar
            size = 4) +   # Slightly larger text as there are fewer bars
  
  # Use a professional, colorblind-friendly color scale
  scale_fill_viridis_d(option = "D") +
  
  # Adjust y-axis to make space for the labels on top
  scale_y_continuous(expand = expansion(mult = c(0, .15))) +
  
  # Add informative labels and a title reflecting the subset
  labs(
    title = "Distribution of Major Splicing Event Types",
    subtitle = paste("Showing", sum(plot_data_filtered$count), "filtered events"),
    x = "Splicing Event Type",
    y = "Number of Events"
  ) +
  
  # Apply a clean, professional theme
  theme_classic(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b = 15)),
    axis.text.x = element_text(face = "bold", size = 12), # No need to angle with 4 bars
    axis.title.y = element_text(face = "bold"),
    axis.title.x = element_text(face = "bold", margin = margin(t = 10)),
    legend.position = "none" # Remove the legend as bars are clearly labeled
  )
print(all_events)

ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_all_types_mattieu.png"), 
  
  # 2. Specify which plot object to save
  plot = all_events,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```
## Unique Genes

```{r}
library(ggplot2)
library(dplyr)
library(forcats)

selected_events <- c("S", "IR", "Alt3", "Alt5")

# --- Prepare the data: count unique genes per splicing type ---
plot_data_genes <- output_transform_all %>%
  filter(COMPLEXAS %in% selected_events) %>%
  distinct(Gene, COMPLEXAS) %>%   # ensure unique gene-event pairs
  count(COMPLEXAS, name = "unique_genes") %>%
  mutate(
    percentage = (unique_genes / sum(unique_genes)) * 100,
    COMPLEXAS = fct_reorder(COMPLEXAS, unique_genes, .desc = TRUE)
  )

print(plot_data_genes)

# --- Plot: distribution of unique genes per event type ---
gene_plot <- ggplot(plot_data_genes, aes(x = COMPLEXAS, y = unique_genes, fill = COMPLEXAS)) +
  geom_bar(stat = "identity", color = "black", alpha = 0.8) +
  geom_text(aes(label = paste0(unique_genes, "\n(", round(percentage, 1), "%)")), 
            vjust = -0.5, size = 4) +
  scale_fill_viridis_d(option = "D") +
  scale_y_continuous(expand = expansion(mult = c(0, .15))) +
  labs(
    title = "Distribution of Unique Genes by Splicing Event Type",
    subtitle = paste("Showing", sum(plot_data_genes$unique_genes), "unique genes"),
    x = "Splicing Event Type",
    y = "Number of Unique Genes"
  ) +
  theme_classic(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b = 15)),
    axis.text.x = element_text(face = "bold", size = 12),
    axis.title.y = element_text(face = "bold"),
    axis.title.x = element_text(face = "bold", margin = margin(t = 10)),
    legend.position = "none"
  )

print(gene_plot)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_all_types_mattieu_unique.png"), 
  
  # 2. Specify which plot object to save
  plot = gene_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

```





```{r}
library(ggplot2)
library(dplyr)
library(forcats)

selected_events <- c("S", "IR", "Alt3", "Alt5")

plot_data_filtered <- output_transform_all %>%
  filter(COMPLEXAS %in% selected_events) %>%
  mutate(
    psi_group = case_when(
      PSI_difference > 5  ~ "ΔPSI > 5",
      PSI_difference < -5 ~ "ΔPSI < -5",
      TRUE                ~ "-0.1 < ΔPSI < 0.1"
    ),
    # set the order explicitly: IR, S, Alt3, Alt5
    COMPLEXAS = factor(COMPLEXAS, levels = c("IR", "S", "Alt3", "Alt5"))
  ) %>%
  count(COMPLEXAS, psi_group, name = "count") %>%
  group_by(psi_group) %>%
  mutate(percentage = (count / sum(count)) * 100)

facet_plot=ggplot(plot_data_filtered, aes(x = COMPLEXAS, y = count, fill = COMPLEXAS)) +
  geom_bar(stat = "identity", color = "black", alpha = 0.8) +
  geom_text(aes(label = paste0(count, "\n(", round(percentage, 1), "%)")),
            vjust = -0.5, size = 4) +
  scale_y_continuous(expand = expansion(mult = c(0, .15))) +
  scale_fill_manual(values = c(
    "IR"   = "#440154",   # your chosen colors
    "S"    = "#21908C",
    "Alt3" = "#FDE725",
    "Alt5" = "grey70"
  )) +
  facet_wrap(~ psi_group) +
  labs(
    title = "Distribution of Major Splicing Event Types by PSI Difference",
    subtitle = "Faceted by PSI Difference Category",
    x = "Splicing Event Type",
    y = "Number of Events"
  ) +
  theme_classic(base_size = 14) +
  theme(
    plot.title    = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b = 15)),
    axis.text.x   = element_text(face = "bold", size = 12),
    axis.title.y  = element_text(face = "bold"),
    axis.title.x  = element_text(face = "bold", margin = margin(t = 10)),
    legend.position = "none"
  )

ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "facet_all.png"), 
  
  # 2. Specify which plot object to save
  plot = facet_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```




## Unique Genes facet



```{r}
library(ggplot2)
library(dplyr)
library(forcats)

selected_events <- c("S", "IR", "Alt3", "Alt5")

# --- Prepare data: count unique genes per COMPLEXAS and PSI group ---
plot_data_genes <- output_transform_all %>%
  filter(COMPLEXAS %in% selected_events) %>%
  mutate(
    psi_group = case_when(
      PSI_difference > 5  ~ "ΔPSI > 5",
      PSI_difference < -5 ~ "ΔPSI < -5",
      TRUE                ~ "-0.1 < ΔPSI < 0.1"
    ),
    COMPLEXAS = factor(COMPLEXAS, levels = c("IR", "S", "Alt3", "Alt5"))
  ) %>%
  distinct(Gene, COMPLEXAS, psi_group) %>%  # ensure unique genes per group
  count(COMPLEXAS, psi_group, name = "unique_genes") %>%
  group_by(psi_group) %>%
  mutate(percentage = (unique_genes / sum(unique_genes)) * 100)

# --- Faceted bar plot for unique genes ---
facet_genes_plot <- ggplot(plot_data_genes, aes(x = COMPLEXAS, y = unique_genes, fill = COMPLEXAS)) +
  geom_bar(stat = "identity", color = "black", alpha = 0.8) +
  geom_text(aes(label = paste0(unique_genes, "\n(", round(percentage, 1), "%)")),
            vjust = -0.5, size = 4) +
  scale_y_continuous(expand = expansion(mult = c(0, .15))) +
  scale_fill_manual(values = c(
    "IR"   = "#440154",
    "S"    = "#21908C",
    "Alt3" = "#FDE725",
    "Alt5" = "grey70"
  )) +
  facet_wrap(~ psi_group) +
  labs(
    title = "Distribution of Unique Genes by Splicing Event Type and PSI Difference",
    subtitle = "Faceted by PSI Difference Category",
    x = "Splicing Event Type",
    y = "Number of Unique Genes"
  ) +
  theme_classic(base_size = 14) +
  theme(
    plot.title    = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b = 15)),
    axis.text.x   = element_text(face = "bold", size = 12),
    axis.title.y  = element_text(face = "bold"),
    axis.title.x  = element_text(face = "bold", margin = margin(t = 10)),
    legend.position = "none"
  )

print(facet_genes_plot)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "facet_all_upset.png"), 
  
  # 2. Specify which plot object to save
  plot = facet_genes_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

```

### Events per gene facet




```{r}
library(ggplot2)
library(dplyr)

# Filter IR events and prepare PSI groups
ir_hist_data <- output_transform_all %>%
  filter(COMPLEXAS == "IR") %>%
  mutate(
    psi_group = case_when(
      PSI_difference > 5  ~ "ΔPSI > 5",
      PSI_difference < -5 ~ "ΔPSI < -5",
      TRUE                ~ "-0.1 < ΔPSI < 0.1"
    )
  ) %>%
  # Count number of IR events per gene within each PSI group
  group_by(psi_group, Gene) %>%
  summarise(n_events = n(), .groups = "drop") %>%
  # Bin counts: 1,2,3,4,5+
  mutate(n_events_bin = case_when(
    n_events >= 5 ~ "5+",
    TRUE          ~ as.character(n_events)
  ))

# Summarize for histogram: number of genes per bin AND total events per bin
hist_data <- ir_hist_data %>%
  group_by(psi_group, n_events_bin) %>%
  summarise(
    n_genes = n(),                       # number of genes in this bin
    total_events = sum(ifelse(n_events >= 5, n_events, n_events)), # total events
    .groups = "drop"
  ) %>%
  mutate(
    n_events_bin = factor(n_events_bin, levels = c("1","2","3","4","5+"))
  )

# --- Plot ---
hist_plot <- ggplot(hist_data, aes(x = n_events_bin, y = n_genes)) +
  geom_col(fill = "#440154", color = "black", alpha = 0.8) +
  geom_text(aes(label = total_events), vjust = -0.5, size = 4) + # absolute event numbers on top
  facet_wrap(~ psi_group) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  labs(
    title = "Distribution of IR Events per Gene by PSI Group",
    x = "Number of IR Events per Gene",
    y = "Number of Genes"
  ) +
  theme_classic(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    strip.text = element_text(face = "bold", size = 12),
    axis.text.x = element_text(face = "bold", size = 12),
    axis.text.y = element_text(face = "bold", size = 12),
    axis.title = element_text(face = "bold")
  )

print(hist_plot)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "facet_all_upset_unique.png"), 
  
  # 2. Specify which plot object to save
  plot = hist_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```


## Density plots for each Alternative Splicing Event


```{r}
## 1. Load required libraries
# Make sure you have them installed: install.packages(c("ggplot2", "dplyr"))
library(ggplot2)
library(dplyr)

# 2. Filter your data for the required event types
# This step remains the same.
filtered_data <- output_transform_all %>%
  filter(COMPLEXAS %in% c("S", "IR", "Alt3", "Alt5"))

# 3. Generate the Smooth Density Plot
# This creates the smooth density background with semi-transparent points on top.
smooth_density_plot <- ggplot(
  data = filtered_data, 
  aes(x = Young_PSI, y = Senescence_PSI)
) +
  
  # --- Layer 1: The Smooth Density Background ---
  # This calculates the density and draws it as a smooth, colored surface.
  stat_density_2d(aes(fill = after_stat(level)), geom = "polygon", alpha = 0.8) +
  
  # --- Layer 2: The Semi-Transparent Points ---
  # This plots all the original dots but makes them transparent
  # so they don't hide the density layer or each other.
  geom_point(shape = 16, size = 1, alpha = 0.1, color = "black") +
  
  # Set the color gradient for the density. Darker = higher density.
  scale_fill_gradientn(colors = c("lightskyblue", "royalblue", "navy")) +
  
  # --- Layer 3: The Diagonal Reference Line ---
  geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +
  
  # Create the 2x2 grid
  facet_wrap(~ COMPLEXAS, nrow = 2) +
  
  # Add titles and labels
  labs(
    title = "PSI Value Density: Young vs. Senescence",
    subtitle = "Color intensity indicates the density of points",
    x = "Young PSI",
    y = "Senescence PSI"
  ) +
  
  # Set axes and a clean theme
  coord_cartesian(xlim = c(0, 100), ylim = c(0, 100), expand = FALSE) +
  theme_bw() + # A clean black & white theme works well for this
  theme(
    aspect.ratio = 1,
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    plot.subtitle = element_text(hjust = 0.5, size = 12),
    strip.background = element_rect(fill = "gray90", color = "black"),
    strip.text = element_text(color = "black", size = 12, face = "bold"),
    legend.position = "none" # The legend is not essential for this visual
  )

# 4. Display the final plot
print(smooth_density_plot)

# 5. (Optional) Save the plot
# ggsave("splicing_smooth_density.png", plot = smooth_density_plot, width = 10, height = 9, dpi = 300)
```


```{r}
library(ggplot2)
library(dplyr)

# Filter data
filtered_data <- output_transform_all %>%
  filter(COMPLEXAS %in% c("S", "IR", "Alt3", "Alt5"))

# Function to create scatter plot for a given event type
make_scatter <- function(event_type) {
  ggplot(
    data = filtered_data %>% filter(COMPLEXAS == event_type),
    aes(x = Young_PSI, y = Senescence_PSI)
  ) +
    geom_point(shape = 16, size = 1, alpha = 0.4, color = "black") +
    geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +
    coord_cartesian(xlim = c(0, 100), ylim = c(0, 100), expand = FALSE) +
    labs(
      title = paste(event_type),
      x = "Proliferative PSI",
      y = "Senescence PSI"
    ) +
    theme_bw() +
    theme(
      aspect.ratio = 1,
      plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
      plot.subtitle = element_text(hjust = 0.5, size = 20)
    )
}

# Generate one plot per event type
plot_S     <- make_scatter("S")
plot_IR    <- make_scatter("IR")
plot_Alt3  <- make_scatter("Alt3")
plot_Alt5  <- make_scatter("Alt5")

# Print plots
print(plot_S)
print(plot_IR)
print(plot_Alt3)
print(plot_Alt5)

# (Optional) Save plots
# ggsave("scatter_S.png", plot = plot_S, width = 5, height = 5, dpi = 300)
# ggsave("scatter_IR.png", plot = plot_IR, width = 5, height = 5, dpi = 300)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "plot_IR.png"), 
  
  # 2. Specify which plot object to save
  plot = plot_IR,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "plot_S.png"), 
  
  # 2. Specify which plot object to save
  plot = plot_S,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```




## Density for Estimate



```{r}
# Make sure your 'data1' dataframe is already loaded in your R session

# --- Improved Article-Style Plot ---
ggplot(data = data1, aes(x = Estimate)) +
  geom_density(
    fill = "gray80",      # Use a light gray for the fill
    color = "black",      # Use a solid black line for high contrast
    alpha = 0.8,          # Keep a slight transparency
    linewidth = 0.8         # Make the line slightly thicker
  ) +
  
  # The theme is where the biggest improvement happens
  theme_bw() + 
  
  # Customize theme elements for a publication look
  theme(
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"), 
    axis.title = element_text(size = 12),           
    axis.text = element_text(size = 10),            
    panel.grid.major = element_line(color = "gray90"), 
    panel.grid.minor = element_blank()              
  ) +
  
  # Your labels (title is unchanged)
  labs(
    title = "Density Plot of Estimate",
    x = "Estimate",
    y = "Density"
  )
```




## Same for deltaPSI



```{r}
ggplot(data = data1, aes(x = .data[["PSI_difference"]])) +
  
  # Density layer with a professional grayscale theme
  geom_density(
    fill = "gray80",      # Light gray fill
    color = "black",      # Solid black outline
    alpha = 0.8,
    linewidth = 0.8         # Slightly thicker line for clarity
  ) +
  
  
  # Start with the clean classic theme
  theme_classic() +
  
  # Add customizations for a polished, article-style look
  theme(
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    plot.subtitle = element_text(hjust = 0.5, size = 12),
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 10, color = "black"),
    axis.line = element_line(color = "black") 
  ) +
  
  # Your original labels and title
  labs(
    title = "Distribution of Splicing Changes in Senescence",
    subtitle = "",
    x = "ΔPSI (PSI Senescence - PSI Proliferative)",
    y = "Density of Events"
  ) +
  
  # Your original zoom setting
  coord_cartesian(xlim = c(-7.5, 7.5))
```


```{r}
library(ggplot2)
library(dplyr)
library(tidyr)

# Reshape into long format
data_long <- data1 %>%
  select(Estimate, PSI_difference) %>%
  pivot_longer(cols = c(Estimate, PSI_difference),
               names_to = "Variable",
               values_to = "Value")

# Compute medians for each variable
medians <- data_long %>%
  group_by(Variable) %>%
  summarize(med = median(Value, na.rm = TRUE))

# Plot with medians shown as line + label
density_plots=ggplot(data_long, aes(x = Value)) +
  geom_density(
    fill = "gray80",
    color = "black",
    alpha = 0.8,
    linewidth = 0.8
  ) +
  geom_vline(data = medians, aes(xintercept = med),
             color = "red", linetype = "dashed", linewidth = 0.9) +
  geom_text(data = medians,
            aes(x = med, y = 0, 
                label = paste0("Median = ", round(med, 2))),
            color = "red", angle = 90, vjust = -0.5, hjust = 0) +
  facet_wrap(~Variable, ncol = 1, scales = "free_x") +
  theme_bw() +
  theme(
    strip.text = element_text(size = 14, face = "bold"),
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 10),
    panel.grid.major = element_line(color = "gray90"),
    panel.grid.minor = element_blank()
  ) +
  labs(
    title = "Parallel Density Distributions with Medians",
    x = NULL,
    y = "Density"
  )



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "density_plots_all.png"), 
  
  # 2. Specify which plot object to save
  plot = density_plots,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```


### statistical




```{r}
library(ggplot2)
library(dplyr)
library(tidyr)

# Reshape into long format
data_long <- data1 %>%
  select(Estimate, PSI_difference) %>%
  pivot_longer(cols = c(Estimate, PSI_difference),
               names_to = "Variable",
               values_to = "Value")

# Compute medians + % positive + Wilcoxon p-value
stats <- data_long %>%
  group_by(Variable) %>%
  summarize(
    med = median(Value, na.rm = TRUE),
    perc_pos = mean(Value > 0, na.rm = TRUE) * 100,
    p_val = wilcox.test(Value, mu = 0, alternative = "greater")$p.value
  ) %>%
  mutate(
    label = paste0("Median = ", round(med, 2),
                   "\n", round(perc_pos, 1), "% > 0",
                   "\n", "p = ", signif(p_val, 3))
  )

# Plot with median line + annotation box
plot_density=ggplot(data_long, aes(x = Value)) +
  geom_density(
    fill = "gray80",
    color = "black",
    alpha = 0.8,
    linewidth = 0.8
  ) +
  geom_vline(data = stats, aes(xintercept = med),
             color = "red", linetype = "dashed", linewidth = 0.9) +
  facet_wrap(~Variable, ncol = 1, scales = "free_x") +
  # Add annotation box in top right of each facet
  geom_text(
    data = stats,
    aes(x = Inf, y = Inf, label = label),
    hjust = 1.1, vjust = 1.1,
    color = "black",
    size = 4,
    fontface = "bold"
  ) +
  theme_bw() +
  theme(
    strip.text = element_text(size = 14, face = "bold"),
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 10, color = "black"),
    panel.grid.major = element_line(color = "gray90"),
    panel.grid.minor = element_blank()
  ) +
  labs(
    title = "Parallel Density Distributions with Medians, % Positive, and Significance",
    x = NULL,
    y = "Density"
  )




ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "density_estimate_psi.png"), 
  
  # 2. Specify which plot object to save
  plot = plot_density,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)





```

THIS ONEEEEEEEEEEEEEEE

```{r}
library(ggplot2)
library(dplyr)
library(tidyr)

# --- Reshape into long format ---
data_long <- data1 %>%
  select(Estimate, PSI_difference) %>%
  pivot_longer(cols = c(Estimate, PSI_difference),
               names_to = "Variable",
               values_to = "Value")

# --- Compute stats (full data) ---
stats <- data_long %>%
  group_by(Variable) %>%
  summarize(
    med = median(Value, na.rm = TRUE),
    perc_pos = mean(Value > 0, na.rm = TRUE) * 100,
    p_val = wilcox.test(Value, mu = 0, alternative = "greater")$p.value
  ) %>%
  mutate(
    label = paste0("Median = ", round(med, 2),
                   "\n", round(perc_pos, 1), "% > 0",
                   "\n", "p = ", signif(p_val, 3))
  )

# --- Plot ---
plot_density <- ggplot(data_long, aes(x = Value)) +
  geom_density(
    fill = "gray80",
    color = "black",
    alpha = 0.8,
    linewidth = 0.8
  ) +
  geom_vline(data = stats, aes(xintercept = med),
             color = "red", linetype = "dashed", linewidth = 0.9) +
  facet_wrap(~Variable, ncol = 1, scales = "free_x") +
  geom_text(
    data = stats,
    aes(x = Inf, y = Inf, label = label),
    hjust = 1.1, vjust = 1.1,
    color = "black",
    size = 4,
    fontface = "bold"
  ) +
  # Zoom PSI_difference facet only
  coord_cartesian(xlim = c(-3, 3)) +  # this will affect all facets
  theme_bw() +
  theme(
    strip.text = element_text(size = 14, face = "bold"),
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 10, color = "black"),
    panel.grid.major = element_line(color = "gray90"),
    panel.grid.minor = element_blank()
  ) +
  labs(
    title = "Parallel Density Distributions with Medians, % Positive, and Significance",
    x = NULL,
    y = "Density"
  )

print(plot_density)


```



## Events per gene

```{r}
events_per_gene_up <- data1 %>%
  filter(PSI_difference > 5) %>%
  count(Gene, name = "num_events")


events_pergene_up=ggplot(events_per_gene_up, aes(x = num_events)) +
  geom_histogram(binwidth = 0.5, fill = "#440154", color = "black") +
  theme_minimal() +
  labs(title = "Distribution of Events per Gene with deltaPSI more than 5",
       x = "Number of Events per Gene",
       y = "Number of Genes")


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_per_gene_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = events_pergene_up,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

events_per_gene_down <- data1 %>%
  filter(PSI_difference < -5) %>%
  count(Gene, name = "num_events")


events_pergene_down=ggplot(events_per_gene_down, aes(x = num_events)) +
  geom_histogram(binwidth = 0.5, fill = "#21908C", color = "black") +
  theme_minimal() +
  labs(title = "Distribution of Events per Gene with deltaPSI more than 5",
       x = "Number of Events per Gene",
       y = "Number of Genes")


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_per_gene_down_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = events_pergene_down,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```



## PSI Ratio Plots

```{r}
correlation_plot <- ggplot(output_transform_all_IR, aes(x = Young_PSI, y = Senescence_PSI)) +
  geom_point(alpha = 0.7,color = "grey45") +  # Density-based point coloring
  geom_smooth(method = "lm", color = "#e31a1c", se = FALSE) +  # Linear regression line
  geom_abline(slope = 1, intercept = 0, color = "black") +  # Line with slope = 1
  labs(title = "PSI distribution for IR Events with PTC",
       x = "Mean Proliferative PSI",
       y = "Mean Senescence PSI") +
  theme_minimal() + # Optional, to limit the x-axis range
  scale_color_viridis_c() +
  geom_point(data = ptc_genes,
             aes(x = Young_PSI, y = Senescence_PSI), 
             color = "red", 
             alpha = 0.7)

```

## Barplot for PTC groups 


```{r}
# Create a column wether the data has a PTC or not
output_transform_all_IR$PTC_Status <- ifelse(output_transform_all_IR$ID %in% ptcs$Event, "PTC", "No PTC")

data_summary <- output_transform_all_IR %>%
  mutate(deltaPSI_group = ifelse(PSI_difference > 0, "deltaPSI > 0", "deltaPSI < 0")) %>%
  group_by(deltaPSI_group, PTC_Status) %>%
  summarise(count = n()) %>%
  ungroup()

total_counts <- output_transform_all_IR %>%
  mutate(deltaPSI_group = ifelse(PSI_difference > 0, "deltaPSI > 0", "deltaPSI < 0")) %>%
  group_by(deltaPSI_group) %>%
  summarise(PTC_Status = "Total", count = n()) %>%
  ungroup()

data_plot <- bind_rows(data_summary, total_counts)

ggplot(data_plot, aes(x = deltaPSI_group, y = count, fill = PTC_Status)) +
  geom_bar(stat = "identity", position = "dodge", width = 0.4) +  
  geom_text(aes(label = count), position = position_dodge(width = 0.4), vjust = -0.5) +  
  labs(x = "Delta PSI Group", y = "Count", fill = "PTC Status") +
  scale_fill_manual(values = c("PTC" = "#e31a1c", "No PTC" = "grey45", "Total" = "grey20")) +    
  theme_minimal(base_size = 12) +  
  theme(axis.text.x = element_text(angle = 45, hjust = 1),  
        legend.position = "top")  


ggsave("C:\\Users\\35196\\Desktop\\Plots\\bar_count_ptc.png", width = 8, height = 6, dpi = 300, type = "cairo")
```


## GO Enrichment Analysis

```{r}

up_genes=data1 %>%
  filter (PSI_difference > 5) 
  
up_genes=unique(up_genes$Gene)

down_genes=data1 %>%
  filter (PSI_difference < -5) 
  
down_genes=unique(down_genes$Gene)


up_genes_entrez <- mapIds(org.Hs.eg.db, keys=up_genes, column="ENTREZID", keytype="SYMBOL", multiVals="first")
down_genes_entrez <- mapIds(org.Hs.eg.db, keys=down_genes, column="ENTREZID", keytype="SYMBOL", multiVals="first")

up_genes_entrez <- na.omit(up_genes_entrez)
down_genes_entrez <- na.omit(down_genes_entrez)


ego1_BP <- enrichGO(
  gene          = up_genes_entrez,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff  = 0.05,
  qvalueCutoff  = 0.05
)

bar_up=barplot(ego1_BP, showCategory = 20) +
  scale_fill_gradient(low = "blue", high = "red") +  
  theme_bw()+
  ggtitle("Top 20 Biological Processes Enriched with deltaPSI > 5")+
  theme(
    plot.title = element_text(size=18, face="bold", hjust=0.5, color="black"),
    axis.title = element_text(size=14, face="bold"),
    axis.text = element_text(size=10, color="black"),
    panel.grid.major = element_line(color="gray80", linetype="dashed"),
    panel.grid.minor = element_blank()
  )

# The dotplot is the best-in-class replacement for the bar plot
go_dot=dotplot(ego1_BP, showCategory = 20, font.size = 10) +
  ggtitle("Top 20 Enriched Biological Processes (Up-regulated)") +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.y = element_text(size = 11)
  )


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "go_mattieu_up_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = go_dot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



ego1_BP <- enrichGO(
  gene          = down_genes_entrez,
  OrgDb         = org.Hs.eg.db,
  keyType       = "ENTREZID",
  ont           = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff  = 0.05,
  qvalueCutoff  = 0.05
)

bar_up=barplot(ego1_BP, showCategory = 20) +
  scale_fill_gradient(low = "blue", high = "red") +  
  theme_bw()+
  ggtitle("Top 20 Biological Processes Enriched with deltaPSI > 5")+
  theme(
    plot.title = element_text(size=18, face="bold", hjust=0.5, color="black"),
    axis.title = element_text(size=14, face="bold"),
    axis.text = element_text(size=10, color="black"),
    panel.grid.major = element_line(color="gray80", linetype="dashed"),
    panel.grid.minor = element_blank()
  )

# The dotplot is the best-in-class replacement for the bar plot
go_dot=dotplot(ego1_BP, showCategory = 20, font.size = 10) +
  ggtitle("Top Enriched Biological Processes (Down-regulated)") +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.y = element_text(size = 11)
  )

print(go_dot)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "go_mattieu_down_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = go_dot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```




## Data 1 Length


```{r}
length_data1_up=ggplot(subset(data1, PSI_difference > 5), aes(x = log(LENGTH))) +
  geom_histogram(bins = 30, fill = "#440154", color = "black") +
  labs(x = "Log(Intron Length)", y = "Count", title = "Intron Length events with deltaPSI > 5") +
  theme_minimal()


print(length_data1_up)

length_data1_down=ggplot(subset(data1, PSI_difference < -5), aes(x = log(LENGTH))) +
  geom_histogram(bins = 30, fill = "#21908C", color = "black") +
  labs(x = "Log(Intron Length)", y = "Count", title = "Intron Length events with deltaPSI < - 5") +
  theme_minimal()

length_data1_low=ggplot(subset(data1, PSI_difference < 0.1 & PSI_difference > -0.1), aes(x = log(LENGTH))) +
  geom_histogram(bins = 30, fill = "#21908C", color = "black") +
  labs(x = "Log(Intron Length)", y = "Count", title = "Intron Length events with deltaPSI between -0.1 and 0.1") +
  theme_minimal()

```



```{r}

# --- Stats ---
kruskal_res <- kruskal.test(log(LENGTH) ~ psi_group, data = plot_data_length)
pairwise_res <- pairwise.wilcox.test(
  plot_data_length$LENGTH,
  plot_data_length$psi_group
)

kruskal_res
pairwise_res

```




```{r}
library(ggplot2)
library(dplyr)
library(viridis)

# --- Prepare your data (This part is unchanged and uses YOUR data1) ---
plot_data_length <- data1 %>%
  mutate(psi_group = case_when(
    PSI_difference > 5                           ~ "ΔPSI > 5",
    PSI_difference < -5                          ~ "ΔPSI < -5",
    PSI_difference < 0.1 & PSI_difference > -0.1 ~ "0.1 < ΔPSI < 0.1 ",
    TRUE                                         ~ NA_character_
  )) %>%
  filter(!is.na(psi_group), !is.na(LENGTH)) %>%
  mutate(psi_group = factor(
    psi_group,
    levels = c("ΔPSI > 5", "ΔPSI < -5", "0.1 < ΔPSI < 0.1 ")
  ))

# --- Median (log scale) ---
median_lengths <- plot_data_length %>%
  group_by(psi_group) %>%
  summarise(median_log_length = median(log(LENGTH), na.rm = TRUE), .groups = 'drop') %>%
  arrange(psi_group) %>%
  
  # --- MODIFICATION IS HERE ---
  # Increased the 'by' value from -0.05 to -0.07 to add more vertical space
  mutate(y_position = seq(from = 0.95 * max(density(log(plot_data_length$LENGTH))$y), 
                          by = -0.07 * max(density(log(plot_data_length$LENGTH))$y), # This value was changed
                          length.out = n()))


# --- Plot densities with medians ---
length_plot <- ggplot(plot_data_length, aes(x = log(LENGTH), fill = psi_group)) +
  geom_density(alpha = 0.6, color = "black", linewidth = 0.2) +
  
  # Dashed vertical lines (no legend)
  geom_vline(
    data = median_lengths,
    aes(xintercept = median_log_length, color = psi_group),
    linetype = "dashed", linewidth = 1, show.legend = FALSE
  ) +
  
  # Median labels in top-right corner
  geom_text(
    data = median_lengths,
    aes(x = Inf, y = y_position, 
        label = paste(psi_group, "=", round(median_log_length, 2)),
        color = psi_group),
    inherit.aes = FALSE,
    hjust = 1.05,
    vjust = 1.2,
    size = 4, fontface = "bold",
    show.legend = FALSE
  ) +
  
  # --- MANUAL COLORS HERE ---
  scale_fill_manual(values = c(
    "ΔPSI > 5" = "#440154",
    "ΔPSI < -5" = "#21908C",
    "0.1 < ΔPSI < 0.1 " = "grey70"
  )) +
  scale_color_manual(values = c(
    "ΔPSI > 5" = "#440154",
    "ΔPSI < -5" = "#21908C",
    "0.1 < ΔPSI < 0.1 " = "grey70"
  )) +
  
  labs(
    title = "Intron Length Distribution by Change in PSI",
    subtitle = "Comparing High, Low, and Near-Zero ΔPSI Groups",
    x = "Log(Length)",
    y = "Density",
    fill = "PSI Group"
  ) +
  
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 11, margin = margin(b = 10)),
    legend.position = "bottom",
    axis.title = element_text(face = "bold"),
    panel.grid.minor = element_blank()
  )

print(length_plot)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "length_mattieu.png"), 
  
  # 2. Specify which plot object to save
  plot = length_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```

### stats file


```{r}
library(dplyr)
library(tidyr)

# --- Function to extract stats for this new dataset ---
extract_stats_v2 <- function(df, dataset_name) {
  
  # Define psi groups (slightly different threshold from first function)
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference < 0.1 & PSI_difference > -0.1  ~ "Near Zero (|ΔPSI| < 0.1)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(LENGTH)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)")
    ))
  
  # Kruskal-Wallis test
  kruskal_res <- kruskal.test(log(LENGTH) ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$LENGTH,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Convert pairwise results to tidy dataframe
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # Medians
  medians <- plot_data_length %>%
    group_by(psi_group) %>%
    summarise(median_length = median(LENGTH, na.rm = TRUE), .groups = "drop")
  
  # Combine everything
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_length) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_length) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Run the function on your dataset (data1 in this example) ---
stats_v2 <- extract_stats_v2(data1, "data1")

# --- Save CSV ---
write.csv(stats_v2, "C:\\Users\\35196\\Desktop\\Thesis_Files\\Supplement_Files\\length_data1_main.csv", row.names = FALSE)

# View the result
stats_v2

```




## Data 1 GC

```{r}
# --- 1. SETUP ---
# install.packages("rstatix") # Run once if needed
library(ggplot2)
library(dplyr)
library(rstatix)
library(viridis)
library(scales) # for the pvalue formatter

# --- 2. PREPARE DATA ---
# Using the same filter as the original GC plot to only compare High vs Low
plot_data_gc <- data6 %>%
  mutate(psi_group = case_when(
    PSI_difference > 5   ~ "High ΔPSI (> 5)",
    PSI_difference < -5  ~ "Low ΔPSI (< -5)",
    TRUE                 ~ NA_character_
  )) %>%
  filter(!is.na(psi_group), !is.na(GC_Ratio)) %>%
  mutate(psi_group = factor(psi_group, levels = c("Low ΔPSI (< -5)", "High ΔPSI (> 5)")))

# --- 3. PERFORM STATS ---
# Calculate the Wilcoxon test for the p-value
stat_test <- plot_data_gc %>%
  wilcox_test(GC_Ratio ~ psi_group) %>%
  add_significance("p")

# Calculate the effect size (rank-biserial correlation)
effect_size <- plot_data_gc %>%
  wilcox_effsize(GC_Ratio ~ psi_group)

# --- 4. CREATE PLOT ---
gc_plot=ggplot(plot_data_gc, aes(x = psi_group, y = GC_Ratio, fill = psi_group)) +
  geom_violin(alpha = 0.8, trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA, alpha = 0.5) +
  
  scale_fill_manual(values = c("Low ΔPSI (< -5)" = "#21908C", "High ΔPSI (> 5)" = "#440154")) +
  
  # Add the statistical summary as a caption for a clean look
  labs(
    caption = paste(
      "Wilcoxon Test:",
      paste0("p = ", scales::pvalue(stat_test$p), " (", stat_test$p.signif, ");"),
      paste0("Effect Size (rbc) = ", round(effect_size$effsize, 2), " (", effect_size$magnitude, ")")
    )
  ) +
  
  labs(
    title = "GC Content by ΔPSI Group with Effect Size",
    subtitle = "Comparing High (ΔPSI > 5) vs. Low (ΔPSI < -5) events in 'data1'",
    x = "",
    y = "GC Ratio"
  ) +
  
  coord_cartesian(clip = "off") + # Allows annotations outside the plot area
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 10)),
    legend.position = "none",
    axis.title = element_text(face = "bold"),
    plot.caption = element_text(hjust = 0.5, vjust = 6, size = 11, face = "italic", color = "black")
  )

print(gc_plot)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "gc_mattieu_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = gc_plot,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

```

```{r}
# --- Prepare data with Near Zero ---
plot_data_gc <- data1 %>%
  mutate(psi_group = case_when(
    PSI_difference > 5                           ~ "ΔPSI > 5",
    PSI_difference < -5                          ~ "ΔPSI < -5",
    PSI_difference < 0.1 & PSI_difference > -0.1 ~ "0.1 < ΔPSI < 0.1 ",
    TRUE                                         ~ NA_character_
  )) %>%
  filter(!is.na(psi_group), !is.na(GC_Ratio)) %>%
  mutate(psi_group = factor(
    psi_group,
    levels = c("ΔPSI < -5", "0.1 < ΔPSI < 0.1 ", "ΔPSI > 5")
  ))

# --- Plot GC content ---
gc_plot <- ggplot(plot_data_gc, aes(x = psi_group, y = GC_Ratio, fill = psi_group)) +
  geom_violin(alpha = 0.8, trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA, alpha = 0.5) +
  
  # Manual colors matching the length plot
  scale_fill_manual(values = c(
    "ΔPSI > 5" = "#440154",
    "ΔPSI < -5" = "#21908C",
    "0.1 < ΔPSI < 0.1 " = "grey70"
  )) +
  
  labs(
    title = "GC Content by ΔPSI Group",
    subtitle = "High, Low, and Near-Zero ΔPSI Groups",
    x = "",
    y = "GC Ratio"
  ) +
  
  coord_cartesian(clip = "off") +
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 10)),
    legend.position = "none",
    axis.title = element_text(face = "bold")
  )

print(gc_plot)


```

```{r}
library(ggpubr)

# --- Define pairwise comparisons ---
my_comparisons <- list(
  c("ΔPSI < -5", "0.1 < ΔPSI < 0.1 "),
  c("ΔPSI > 5", "0.1 < ΔPSI < 0.1 "),
  c("ΔPSI < -5", "ΔPSI > 5")
)

# --- GC content plot with significance stars ---
gc_plot <- ggplot(plot_data_gc, aes(x = psi_group, y = GC_Ratio, fill = psi_group)) +
  geom_violin(alpha = 0.8, trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA, alpha = 0.5) +
  
  # Pairwise Wilcoxon tests
  stat_compare_means(
    comparisons = my_comparisons,
    method = "wilcox.test",
    label = "p.signif",
    size = 6
  ) +
  
  scale_fill_manual(values = c(
    "ΔPSI > 5" = "#440154",
    "ΔPSI < -5" = "#21908C",
    "0.1 < ΔPSI < 0.1 " = "grey70"
  )) +
  
  labs(
    title = "GC Content by ΔPSI Group",
    subtitle = "Pairwise Wilcoxon tests across Low, Near Zero, and High ΔPSI",
    x = "",
    y = "GC Ratio"
  ) +
  
  coord_cartesian(clip = "off") +
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 10)),
    legend.position = "none",
    axis.title = element_text(face = "bold")
  )

print(gc_plot)

```

```{r}
# Kruskal-Wallis test
kruskal_gc <- kruskal.test(GC_Ratio ~ psi_group, data = plot_data_gc)

# Pairwise Wilcoxon tests with BH correction
pairwise_gc <- pairwise.wilcox.test(
  plot_data_gc$GC_Ratio,
  plot_data_gc$psi_group,
  p.adjust.method = "BH"
)

# Print results
kruskal_gc
pairwise_gc

```

### stats file

```{r}
library(dplyr)
library(tidyr)

# --- Function to extract stats for this new dataset ---
extract_stats_v2 <- function(df, dataset_name) {
  
  # Define psi groups (slightly different threshold from first function)
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference < 0.1 & PSI_difference > -0.1  ~ "Near Zero (|ΔPSI| < 0.1)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(GC_Ratio)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)")
    ))
  
  # Kruskal-Wallis test
  kruskal_res <- kruskal.test(GC_Ratio ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$GC_Ratio,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Convert pairwise results to tidy dataframe
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # Medians
  medians <- plot_data_length %>%
    group_by(psi_group) %>%
    summarise(median_gc = median(GC_Ratio, na.rm = TRUE), .groups = "drop")
  
  # Combine everything
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_gc) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_gc) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Run the function on your dataset (data1 in this example) ---
stats_v2 <- extract_stats_v2(data1, "data1")


# --- Save CSV ---
write.csv(stats_v2, "C:\\Users\\35196\\Desktop\\Thesis_Files\\Supplement_Files\\gc_data1_main.csv", row.names = FALSE)

# View the result
stats_v2

```







## Splice site strength


```{r}
# --- 1. SETUP: LOAD LIBRARIES ---
library(ggplot2)
library(dplyr)
library(tidyr)     # For the pivot_longer() function
library(ggpubr)    # For easy statistical comparisons
library(patchwork) # To combine plots

# --- 2. PREPARE THE DATA ---
# This 'plot_data_ss' will be used for all subsequent plots
plot_data_ss <- data1 %>%
  # Define the same PSI groups as before
  mutate(psi_group = case_when(
    PSI_difference > 5                     ~ "High ΔPSI (> 5)",
    PSI_difference < -5                    ~ "Low ΔPSI (< -5)",
    PSI_difference < 0.1 & PSI_difference > -0.1 ~ "Near Zero (|ΔPSI| < 0.1)",
    TRUE                                   ~ NA_character_
  )) %>%
  filter(!is.na(psi_group)) %>%
  
  # The KEY step: pivot the 3_score and 5_score columns into a long format
  pivot_longer(
    cols = c(`3_score`, `5_score`), # Use backticks for names starting with numbers
    names_to = "splice_site_type",
    values_to = "score"
  ) %>%
  
  # Clean up the names for better plot labels
  mutate(
    splice_site_type = if_else(splice_site_type == "3_score", "3' Splice Site", "5' Splice Site"),
    # Set the order of groups for plotting
    psi_group = factor(psi_group, levels = c("Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)", "High ΔPSI (> 5)"))
  ) %>%
  filter(!is.na(score)) # Remove any rows where the score is missing



# Define the comparisons you want to make
my_comparisons <- list(
  c("Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)"), 
  c("High ΔPSI (> 5)", "Near Zero (|ΔPSI| < 0.1)"),
  c("Low ΔPSI (< -5)", "High ΔPSI (> 5)")
)


ss_violin=ggplot(plot_data_ss, aes(x = psi_group, y = score, fill = psi_group)) +
  geom_violin(trim = FALSE, alpha = 0.8) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA) +
  
  # The crucial part: create separate plots for 3' and 5' sites
  # `scales = "free_y"` is ESSENTIAL because their scores are on different scales
  facet_wrap(~ splice_site_type, scales = "free_y") +
  
  # Add statistical tests comparing high/low groups to the baseline
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test", 
                     label = "p.signif", # Use stars for significance
                     size = 6) +

  scale_fill_manual(values = c("#21908C", "grey70", "#440154")) +
  
  labs(
    title = "Splice Site Strength Comparison Across ΔPSI Groups",
    subtitle = "Comparing High and Low ΔPSI events against a Near-Zero baseline",
    x = "PSI Group",
    y = "Splice Site Score"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 15)),
    legend.position = "none",
    axis.text.x = element_text(angle = 30, hjust = 1),
    strip.text = element_text(face = "bold", size = 14) # Facet titles
  )

print(ss_violin)

ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "splice_site_violin_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = ss_violin,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```



```{r}
# 3' Splice Site
kruskal_ss_3 <- kruskal.test(score ~ psi_group, 
                             data = plot_data_ss %>% filter(splice_site_type == "3' Splice Site"))
pairwise_ss_3 <- pairwise.wilcox.test(
  plot_data_ss %>% filter(splice_site_type == "3' Splice Site") %>% pull(score),
  plot_data_ss %>% filter(splice_site_type == "3' Splice Site") %>% pull(psi_group),
  p.adjust.method = "BH"
)

# 5' Splice Site
kruskal_ss_5 <- kruskal.test(score ~ psi_group, 
                             data = plot_data_ss %>% filter(splice_site_type == "5' Splice Site"))
pairwise_ss_5 <- pairwise.wilcox.test(
  plot_data_ss %>% filter(splice_site_type == "5' Splice Site") %>% pull(score),
  plot_data_ss %>% filter(splice_site_type == "5' Splice Site") %>% pull(psi_group),
  p.adjust.method = "BH"
)

# Print results
kruskal_ss_3
pairwise_ss_3

kruskal_ss_5
pairwise_ss_5

```


### stats file


```{r}
library(dplyr)
library(tidyr)

# --- Function to extract stats for this new dataset ---
extract_stats_v2 <- function(df, dataset_name) {
  
  # Define psi groups (slightly different threshold from first function)
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference < 0.1 & PSI_difference > -0.1  ~ "Near Zero (|ΔPSI| < 0.1)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(`3_score`)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)")
    ))
  
  # Kruskal-Wallis test
  kruskal_res <- kruskal.test(`3_score` ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$`3_score`,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Convert pairwise results to tidy dataframe
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # Medians
  medians <- plot_data_length %>%
    group_by(psi_group) %>%
    summarise(median_3 = median(`3_score`, na.rm = TRUE), .groups = "drop")
  
  # Combine everything
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_3) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_3) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Run the function on your dataset (data1 in this example) ---
stats_v2 <- extract_stats_v2(data1, "data1")


write.csv(stats_v2, "C:\\Users\\35196\\Desktop\\Thesis_Files\\Supplement_Files\\3score_data1_main.csv", row.names = FALSE)


```


5

```{r}
library(dplyr)
library(tidyr)

# --- Function to extract stats for this new dataset ---
extract_stats_v2 <- function(df, dataset_name) {
  
  # Define psi groups (slightly different threshold from first function)
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference < 0.1 & PSI_difference > -0.1  ~ "Near Zero (|ΔPSI| < 0.1)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(`5_score`)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)")
    ))
  
  # Kruskal-Wallis test
  kruskal_res <- kruskal.test(`5_score` ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$`5_score`,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Convert pairwise results to tidy dataframe
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # Medians
  medians <- plot_data_length %>%
    group_by(psi_group) %>%
    summarise(median_5 = median(`5_score`, na.rm = TRUE), .groups = "drop")
  
  # Combine everything
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_5) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_5) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Run the function on your dataset (data1 in this example) ---
stats_v2 <- extract_stats_v2(data1, "data1")


write.csv(stats_v2, "C:\\Users\\35196\\Desktop\\Thesis_Files\\Supplement_Files\\5score_data1_main.csv", row.names = FALSE)


```







```{r}
# --- 1. SETUP: LOAD LIBRARIES ---
library(ggplot2)
library(dplyr)
library(tidyr)
library(ggpubr)
library(patchwork)

# --- 2. PREPARE THE DATA ---
plot_data_ss <- data1 %>%
  mutate(psi_group = case_when(
    PSI_difference > 5                     ~ "High ΔPSI (> 5)",
    PSI_difference < -5                    ~ "Low ΔPSI (< -5)",
    PSI_difference < 0.1 & PSI_difference > -0.1 ~ "Near Zero (|ΔPSI| < 0.1)",
    TRUE                                   ~ NA_character_
  )) %>%
  filter(!is.na(psi_group)) %>%
  pivot_longer(
    cols = c(`3_score`, `5_score`),
    names_to = "splice_site_type",
    values_to = "score"
  ) %>%
  mutate(
    splice_site_type = if_else(splice_site_type == "3_score", "3' Splice Site", "5' Splice Site"),
    psi_group = factor(psi_group, levels = c("Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)", "High ΔPSI (> 5)"))
  ) %>%
  filter(!is.na(score))

# --- 3. DEFINE PAIRWISE COMPARISONS ---
my_comparisons <- list(
  c("Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.1)"),
  c("High ΔPSI (> 5)", "Near Zero (|ΔPSI| < 0.1)"),
  c("Low ΔPSI (< -5)", "High ΔPSI (> 5)")
)

# --- 4. CALCULATE EFFECT SIZES MANUALLY ---
compute_r <- function(x, y) {
  # Rank-biserial correlation: r = Z / sqrt(N)
  wt <- wilcox.test(x, y)
  Z <- qnorm(wt$p.value/2, lower.tail = FALSE) # two-sided
  N <- length(x) + length(y)
  r <- Z / sqrt(N)
  return(r)
}

effect_sizes <- data.frame()
for(facet in unique(plot_data_ss$splice_site_type)) {
  sub <- plot_data_ss %>% filter(splice_site_type == facet)
  for(comp in my_comparisons) {
    x <- sub$score[sub$psi_group == comp[1]]
    y <- sub$score[sub$psi_group == comp[2]]
    r <- compute_r(x, y)
    effect_sizes <- rbind(effect_sizes, data.frame(
      splice_site_type = facet,
      group1 = comp[1],
      group2 = comp[2],
      effsize = round(r, 2)
    ))
  }
}

# Position effect size labels below the violins
effect_sizes <- effect_sizes %>%
  group_by(splice_site_type) %>%
  mutate(
    x.position = (as.numeric(factor(group1, levels = levels(plot_data_ss$psi_group))) +
                  as.numeric(factor(group2, levels = levels(plot_data_ss$psi_group)))) / 2,
    y.position = min(plot_data_ss$score[plot_data_ss$splice_site_type == splice_site_type], na.rm = TRUE) -
           0.1 * abs(min(plot_data_ss$score[plot_data_ss$splice_site_type == splice_site_type], na.rm = TRUE)),
    label = paste0("r = ", effsize)
  )

# --- 5. CREATE VIOLIN + BOX PLOT WITHOUT EFFECT SIZE LABELS ---
ss_violin <- ggplot(plot_data_ss, aes(x = psi_group, y = score, fill = psi_group)) +
  geom_violin(trim = FALSE, alpha = 0.8) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA) +
  facet_wrap(~ splice_site_type, scales = "free_y") +
  stat_compare_means(
    comparisons = my_comparisons,
    method = "wilcox.test",
    label = "p.signif",
    size = 6
  ) +
  scale_fill_manual(values = c("#21908C", "grey70", "#440154")) +
  labs(
    title = "Splice Site Strength Comparison Across ΔPSI Groups",
    subtitle = "Pairwise Wilcoxon tests shown",
    x = "PSI Group",
    y = "Splice Site Score"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 15)),
    legend.position = "none",
    axis.text.x = element_text(angle = 30, hjust = 1),
    strip.text = element_text(face = "bold", size = 14)
  )

# --- 6. PRINT THE PLOT ---
print(ss_violin)


```




```{r}
# --- 1. SETUP: LOAD LIBRARIES ---
library(ggplot2)
library(dplyr)
library(tidyr)
library(ggpubr)
library(patchwork)

# --- 2. PREPARE THE DATA ---
plot_data_ss <- data1 %>%
  mutate(psi_group = case_when(
    PSI_difference > 5                     ~ "ΔPSI > 5",
    PSI_difference < -5                    ~ "ΔPSI < -5",
    PSI_difference < 0.1 & PSI_difference > -0.1 ~ "-0.1 < ΔPSI < 0.1",
    TRUE                                   ~ NA_character_
  )) %>%
  filter(!is.na(psi_group)) %>%
  pivot_longer(
    cols = c(`3_score`, `5_score`),
    names_to = "splice_site_type",
    values_to = "score"
  ) %>%
  mutate(
    splice_site_type = if_else(splice_site_type == "3_score", "3' Splice Site", "5' Splice Site"),
    psi_group = factor(psi_group, levels = c("ΔPSI < -5", "-0.1 < ΔPSI < 0.1", "ΔPSI > 5"))
  ) %>%
  filter(!is.na(score))

# --- 3. DEFINE PAIRWISE COMPARISONS ---
my_comparisons <- list(
  c("ΔPSI < -5", "-0.1 < ΔPSI < 0.1"),
  c("ΔPSI > 5", "-0.1 < ΔPSI < 0.1"),
  c("ΔPSI < -5", "ΔPSI > 5")
)

# --- 4. CREATE VIOLIN + BOX PLOT WITH WILCOXON LINES (NO EFFECT SIZE LABELS) ---
ss_violin <- ggplot(plot_data_ss, aes(x = psi_group, y = score, fill = psi_group)) +
  geom_violin(trim = FALSE, alpha = 0.8) +
  geom_boxplot(width = 0.1, fill = "white", outlier.shape = NA) +
  facet_wrap(~ splice_site_type, scales = "free_y") +
  stat_compare_means(
    comparisons = my_comparisons,
    method = "wilcox.test",
    label = "p.signif",
    size = 6
  ) +
  scale_fill_manual(values = c("#21908C", "grey70", "#440154")) +
  labs(
    title = "Splice Site Strength Comparison Across ΔPSI Groups",
    subtitle = "Pairwise Wilcoxon tests shown",
    x = "PSI Group",
    y = "Splice Site Score"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.subtitle = element_text(hjust = 0.5, margin = margin(b = 15)),
    legend.position = "none",
    axis.text.x = element_text(angle = 30, hjust = 1),
    strip.text = element_text(face = "bold", size = 14)
  )

# --- 5. PRINT THE PLOT ---
print(ss_violin)

```




## Save plots





# Data for the second dataset and third dataset for the ribodeplition 

```{r}
data2 <- read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compare_proli_deep.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data3 <- read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compare_proli_quies.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data4 <- read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compare_proli_senesc.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data5 <- read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compare_senescence_deep.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data6 <- read_delim("C:/Users/35196/Desktop/Important_files/casella_files/compareOutput_toPlot.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data7 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compare_quies_senesc.tsv", 
                    delim = "\t", escape_double = FALSE, 
                    trim_ws = TRUE)
data8 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/bj_senescence/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
data9 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/hff_senescence/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
data10 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/imr_senescence/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
data11 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/wi_senescence/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
data12 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_senescence/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

data13 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_pre/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

data14 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/suda/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
data15 = read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/compareOutput_toPlot.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)




data2 <- data2[data2$COMPLEX == "IR", ]
data3 <- data3[data3$COMPLEX == "IR", ]
data4 <- data4[data4$COMPLEX == "IR", ]
data5 <- data5[data5$COMPLEX == "IR", ]
data6 <- data6[data6$COMPLEX == "IR", ]
data7 = data7[data7$COMPLEX == "IR", ]
data8 = data8[data8$COMPLEX == "IR", ]
data9 = data9[data9$COMPLEX == "IR", ]
data10 = data10[data10$COMPLEX == "IR", ]
data11 = data11[data11$COMPLEX == "IR", ]
data12 = data12[data12$COMPLEX == "IR", ]
data13 = data13[data13$COMPLEX == "IR", ]
data14 = data14[data14$COMPLEX == "IR", ]
data15 = data15[data15$COMPLEX == "IR", ]
```



# Data with each feature for differential splicing analysis


For this step, a new file was obtained using a custom python script to get the PSIs and the reads that include and exclude the Intron Retention event


```{r}

#View(output) # this is for the first dataset

output1= output_transform_all_IR
output2= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_proli_deep.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)



output2<- output2[output2$Event == "IR", ]
output2 <- output2[!is.na(output2$Gene), ]
output2$PSI_difference=output2$Deep_PSI-output2$Proliferative_PSI

output3= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_proli_quies.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output3<- output3[output3$Event == "IR", ]
output3 <- output3[!is.na(output3$Gene), ]
output3$PSI_difference=output3$Quiescent_PSI-output3$Proliferative_PSI




output4= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_proli_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output4<- output4[output4$Event == "IR", ]
output4 <- output4[!is.na(output4$Gene), ]
output4$PSI_difference=output4$Senescence_PSI-output4$Proliferative_PSI




output5= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_senesc_deep.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output5<- output5[output5$Event == "IR", ]
output5 <- output5[!is.na(output5$Gene), ]
output5$PSI_difference=output5$Deep_PSI-output5$Senescence_PSI



output6= read_delim("C:/Users/35196/Desktop/Important_files/casella_files/ribo_output.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output6<- output6[output6$Event == "IR", ]
output6 <- output6[!is.na(output6$Gene), ]
output6$PSI_difference=output6$Senescence_PSI-output6$Young_PSI


output7= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_quies_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output7<- output7[output7$Event == "IR", ]
output7 <- output7[!is.na(output7$Gene), ]
output7$PSI_difference=output7$Senescence_PSI-output7$Quiescent_PSI

output8= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/bj_senescence/transformed_bj_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output8<- output8[output8$Event == "IR", ]
output8 <- output8[!is.na(output8$Gene), ]
output8$PSI_difference=output8$Senescence_PSI-output8$Proliferative_PSI


output9= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/hff_senescence/transformed_hff_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output9<- output9[output9$Event == "IR", ]
output9 <- output9[!is.na(output9$Gene), ]
output9$PSI_difference=output9$Senescence_PSI-output9$Proliferative_PSI

output10= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/imr_senescence/transformed_imr_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output10<- output10[output10$Event == "IR", ]
output10 <- output10[!is.na(output10$Gene), ]
output10$PSI_difference=output10$Senescence_PSI-output10$Proliferative_PSI

output11= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/wi_senescence/transformed_wi_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output11<- output11[output11$Event == "IR", ]
output11 <- output11[!is.na(output11$Gene), ]
output11$PSI_difference=output11$Senescence_PSI-output11$Proliferative_PSI



output12= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_senescence/transformed_proli_senescence.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output12<- output12[output12$Event == "IR", ]
output12 <- output12[!is.na(output12$Gene), ]
output12$PSI_difference=output12$Senescence_PSI-output12$Proliferative_PSI


output13= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_pre/transformed_proli_pre.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output13<- output13[output13$Event == "IR", ]
output13 <- output13[!is.na(output13$Gene), ]
output13$PSI_difference=output13$PreSenescence_PSI-output13$Proliferative_PSI


output14= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/suda/transformed_suda.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


output14<- output14[output14$Event == "IR", ]
output14 <- output14[!is.na(output14$Gene), ]
output14$PSI_difference=output14$Senescence_PSI-output14$Proliferative_PSI


output15= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/transformed_quies_deep.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

output15<- output15[output15$Event == "IR", ]
output15 <- output15[!is.na(output15$Gene), ]
output15$PSI_difference=output15$Deep_PSI-output15$Quiescent_PSI


```


Now we merge the deltaPSI values with the principal dataset

```{r}
data2=data2 %>% 
  left_join(output2, by = c("EVENT" = "ID"))

data3=data3 %>% 
  left_join(output3, by = c("EVENT" = "ID"))

data4=data4 %>% 
  left_join(output4, by = c("EVENT" = "ID"))

data5=data5 %>% 
  left_join(output5, by = c("EVENT" = "ID"))

data6=data6 %>% 
  left_join(output6, by = c("EVENT" = "ID"))

data7=data7 %>% 
  left_join(output7, by = c("EVENT" = "ID"))

data8=data8 %>% 
  left_join(output8, by = c("EVENT" = "ID"))

data9=data9 %>% 
  left_join(output9, by = c("EVENT" = "ID"))

data10=data10 %>% 
  left_join(output10, by = c("EVENT" = "ID"))

data11=data11 %>% 
  left_join(output11, by = c("EVENT" = "ID"))

data12=data12 %>% 
  left_join(output12, by = c("EVENT" = "ID"))

data13=data13 %>% 
  left_join(output13, by = c("EVENT" = "ID"))

data14=data14 %>% 
  left_join(output14, by = c("EVENT" = "ID"))

data15=data15 %>% 
  left_join(output15, by = c("EVENT" = "ID"))
```


Now we add more columns in specific to the coordinates of the chromossome the exon start and end


```{r}
library(stringr)
data2 <- data2 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)


data2$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data2$COORD))
data2$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data2$COORD))

data3 <- data3 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)


data3$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data3$COORD))
data3$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data3$COORD))



data4 <- data4 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data4$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data4$COORD))
data4$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data4$COORD))



data5 <- data5 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data5$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data5$COORD))
data5$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data5$COORD))



data6 <- data6 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data6$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data6$COORD))
data6$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data6$COORD))


data7 <- data7 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data7$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data7$COORD))
data7$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data7$COORD))

data8 <- data8 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data8$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data8$COORD))
data8$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data8$COORD))

data9 <- data9 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data9$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data9$COORD))
data9$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data9$COORD))

data10 <- data10 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data10$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data10$COORD))
data10$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data10$COORD))

data11 <- data11 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data11$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data11$COORD))
data11$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data11$COORD))

data12 <- data12 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data12$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data12$COORD))
data12$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data12$COORD))

data13 <- data13 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data13$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data13$COORD))
data13$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data13$COORD))

data14 <- data14 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data14$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data14$COORD))
data14$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data14$COORD))

data15 <- data15 %>%
  mutate(
    CHR = str_extract(FullCO, "chr[\\dXYM]+"),
    Up_Exon_Start = as.numeric(str_extract(FullCO, "(?<=:)(\\d+)(?=-)")),
    Up_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?==)")),
    Down_Exon_Start = as.numeric(str_extract(FullCO, "(?<=\\=)(\\d+)(?=-)")),
    Down_Exon_End = as.numeric(str_extract(FullCO, "(?<=-)(\\d+)(?=:)")),
    Strand = str_extract(FullCO, "(?<=:)[+-]$")
    )%>%
  select(-FullCO)

data15$Intron_Start <- as.numeric(sub(".*:(\\d+)-.*", "\\1", data15$COORD))
data15$Intron_End <- as.numeric(sub(".*-(\\d+)", "\\1", data15$COORD))



```


# Estimate calculation for all datasets


```{r}
log_fun_senes <- function(data) {
  final_results_all <- data.frame(Gene = character(),
                                  Group = character(),
                                  Event = integer(),
                                  Estimate = numeric(),
                                  StdError = numeric(),
                                  ZValue = numeric(),
                                  PValue = numeric(),
                                  EventID = numeric(),
                                  stringsAsFactors = FALSE)
  
  genes <- unique(data$Gene)
  for (gene in genes) {
    gene_data <- data[data$Gene == gene, ]
    
    num_events <- nrow(gene_data) / 4
    
    for (event in seq_len(num_events)) {
      event_data <- gene_data[((event - 1) * 4 + 1):(event * 4), ]
      
      model <- tryCatch(
        glm(cbind(Inclusion, Exclusion) ~ Group, data = event_data, family = binomial(link = "logit")),
        error = function(e) NULL,
        warning = function(w) NULL
      )
      
      if (!is.null(model)) {
        summary_model <- summary(model)
        if ("GroupSenescence" %in% rownames(summary_model$coefficients)) {
          coef_info <- summary_model$coefficients["GroupSenescence", ]
          representative_row <- event_data[1, c("Group")]
          event_id = event_data$ID[1]
          final_results_all <- rbind(final_results_all, data.frame(
            Gene = gene,
            Group = representative_row$Group,
            Event = event,
            Estimate = coef_info[1],
            StdError = coef_info[2],
            ZValue = coef_info[3],
            PValue = coef_info[4],
            EventID = event_id
          ))
        }
      }
    }
  }
  return(final_results_all)
}





log_fun_senes_3 <- function(data) {
  final_results_all <- data.frame(Gene = character(),
                                  Group = character(),
                                  Event = integer(),
                                  Estimate = numeric(),
                                  StdError = numeric(),
                                  ZValue = numeric(),
                                  PValue = numeric(),
                                  EventID = numeric(),
                                  stringsAsFactors = FALSE)
  
  genes <- unique(data$Gene)
  for (gene in genes) {
    gene_data <- data[data$Gene == gene, ]
    
    num_events <- nrow(gene_data) / 6
    
    for (event in seq_len(num_events)) {
      event_data <- gene_data[((event - 1) * 6 + 1):(event * 6), ]
      
      model <- tryCatch(
        glm(cbind(Inclusion, Exclusion) ~ Group, data = event_data, family = binomial(link = "logit")),
        error = function(e) NULL,
        warning = function(w) NULL
      )
      
      if (!is.null(model)) {
        summary_model <- summary(model)
        if ("GroupSenescence" %in% rownames(summary_model$coefficients)) {
          coef_info <- summary_model$coefficients["GroupSenescence", ]
          representative_row <- event_data[1, c("Group")]
          event_id = event_data$ID[1]
          final_results_all <- rbind(final_results_all, data.frame(
            Gene = gene,
            Group = representative_row$Group,
            Event = event,
            Estimate = coef_info[1],
            StdError = coef_info[2],
            ZValue = coef_info[3],
            PValue = coef_info[4],
            EventID = event_id
          ))
        }
      }
    }
  }
  return(final_results_all)
}



log_fun_deep <- function(data) {
  final_results_all <- data.frame(Gene = character(),
                                  Group = character(),
                                  Event = integer(),
                                  Estimate = numeric(),
                                  StdError = numeric(),
                                  ZValue = numeric(),
                                  PValue = numeric(),
                                  EventID = numeric(),
                                  stringsAsFactors = FALSE)
  
  genes <- unique(data$Gene)
  for (gene in genes) {
    gene_data <- data[data$Gene == gene, ]
    
    num_events <- nrow(gene_data) / 4
    
    for (event in seq_len(num_events)) {
      event_data <- gene_data[((event - 1) * 4 + 1):(event * 4), ]
      
      model <- tryCatch(
        glm(cbind(Inclusion, Exclusion) ~ Group, data = event_data, family = binomial(link = "logit")),
        error = function(e) NULL,
        warning = function(w) NULL
      )
      
      if (!is.null(model)) {
        summary_model <- summary(model)
        if ("GroupDeep" %in% rownames(summary_model$coefficients)) {
          coef_info <- summary_model$coefficients["GroupDeep", ]
          representative_row <- event_data[1, c("Group")]
          event_id = event_data$ID[1]
          final_results_all <- rbind(final_results_all, data.frame(
            Gene = gene,
            Group = representative_row$Group,
            Event = event,
            Estimate = coef_info[1],
            StdError = coef_info[2],
            ZValue = coef_info[3],
            PValue = coef_info[4],
            EventID = event_id
          ))
        }
      }
    }
  }
  return(final_results_all)
}



log_fun_quies <- function(data) {
  final_results_all <- data.frame(Gene = character(),
                                  Group = character(),
                                  Event = integer(),
                                  Estimate = numeric(),
                                  StdError = numeric(),
                                  ZValue = numeric(),
                                  PValue = numeric(),
                                  EventID = numeric(),
                                  stringsAsFactors = FALSE)
  
  genes <- unique(data$Gene)
  for (gene in genes) {
    gene_data <- data[data$Gene == gene, ]
    
    num_events <- nrow(gene_data) / 4
    
    for (event in seq_len(num_events)) {
      event_data <- gene_data[((event - 1) * 4 + 1):(event * 4), ]
      
      model <- tryCatch(
        glm(cbind(Inclusion, Exclusion) ~ Group, data = event_data, family = binomial(link = "logit")),
        error = function(e) NULL,
        warning = function(w) NULL
      )
      
      if (!is.null(model)) {
        summary_model <- summary(model)
        if ("GroupQuiescent" %in% rownames(summary_model$coefficients)) {
          coef_info <- summary_model$coefficients["GroupQuiescent", ]
          representative_row <- event_data[1, c("Group")]
          event_id = event_data$ID[1]
          final_results_all <- rbind(final_results_all, data.frame(
            Gene = gene,
            Group = representative_row$Group,
            Event = event,
            Estimate = coef_info[1],
            StdError = coef_info[2],
            ZValue = coef_info[3],
            PValue = coef_info[4],
            EventID = event_id
          ))
        }
      }
    }
  }
  return(final_results_all)
}



```


```{r}
estimate2= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/output_proli_deep.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate2$Group <- factor(estimate2$Group, levels = c("Proliferative", "Deep"))

estimate2<- estimate2[estimate2$Event == "IR", ]
estimate2 <- estimate2[!is.na(estimate2$Gene), ]

estimate3= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/output_proli_quies.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate3$Group <- factor(estimate3$Group, levels = c("Proliferative", "Quiescent"))


estimate3<- estimate3[estimate3$Event == "IR", ]
estimate3 <- estimate3[!is.na(estimate3$Gene), ]




estimate4= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/output_proli_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate4<- estimate4[estimate4$Event == "IR", ]
estimate4 <- estimate4[!is.na(estimate4$Gene), ]




estimate5= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/output_senesc_deep.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate5$Group <- factor(estimate5$Group, levels = c("Senescence", "Deep"))


estimate5<- estimate5[estimate5$Event == "IR", ]
estimate5 <- estimate5[!is.na(estimate5$Gene), ]



estimate6= read_delim("C:/Users/35196/Desktop/Important_files/casella_files/output_all.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)



estimate6$Group[estimate6$Group == "Young"] <- "Proliferative"
estimate6$Group <- factor(estimate6$Group, levels = c("Proliferative", "Senescence"))


estimate6<- estimate6[estimate6$Event == "IR", ]
estimate6 <- estimate6[!is.na(estimate6$Gene), ]


estimate7= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/zhang/intron/output_quies_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate7<- estimate7[estimate7$Event == "IR", ]
estimate7 <- estimate7[!is.na(estimate7$Gene), ]

estimate8= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/bj_senescence/output_bj_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate8<- estimate8[estimate8$Event == "IR", ]
estimate8 <- estimate8[!is.na(estimate8$Gene), ]


estimate9= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/hff_senescence/output_hff_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate9<- estimate9[estimate9$Event == "IR", ]
estimate9 <- estimate9[!is.na(estimate9$Gene), ]

estimate10= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/imr_senescence/output_imr_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate10<- estimate10[estimate10$Event == "IR", ]
estimate10 <- estimate10[!is.na(estimate10$Gene), ]

estimate11= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Marthandan/wi_senescence/output_wi_senesc.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate11<- estimate11[estimate11$Event == "IR", ]
estimate11 <- estimate11[!is.na(estimate11$Gene), ]



estimate12= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_senescence/output_proli_senescence.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate12<- estimate12[estimate12$Event == "IR", ]
estimate12 <- estimate12[!is.na(estimate12$Gene), ]


estimate13= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/purcell/proli_pre/output_proli_pre.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate13<- estimate13[estimate13$Event == "IR", ]
estimate13 <- estimate13[!is.na(estimate13$Gene), ]


estimate14= read_delim("C:/Users/35196/Desktop/Important_files/Comparison_with_all/suda/output_suda.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)


estimate14<- estimate14[estimate14$Event == "IR", ]
estimate14 <- estimate14[!is.na(estimate14$Gene), ]



```


Now we calculate the results from the estimate

This was done in the qds server from GIMM due to the lack of power 

```{r}
results4 = log_fun_senes(estimate4)
results6 = log_fun_senes(estimate6) # não está a dar
results7 = log_fun_senes(estimate7) # not sure
results14 = log_fun_senes(estimate14)
save.image(file = "C:\\Users\\35196\\Desktop\\Important_files\\JoaoR.RData")
```


```{r}
# for marthandan data and purcell
results8 = log_fun_senes_3(estimate8)
results9 = log_fun_senes_3(estimate9)
results10 = log_fun_senes_3(estimate10)
results11 = log_fun_senes_3(estimate11)
results12 = log_fun_senes_3(estimate12)


```


Now for zhang with deep senescence


```{r}
results2 = log_fun_deep(estimate2)
results5 = log_fun_deep(estimate5)

```

And for quiescence

```{r}
results3 = log_fun_quies(estimate3)
```



Now we will save all this files in order to perform the GC Ratio analysis in slurm as well as the Splice Site Strength

```{r}
write.csv(data2, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data2.csv",row.names = FALSE)

write.csv(data3, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data3.csv",row.names = FALSE)

write.csv(data4, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data4.csv",row.names = FALSE)

write.csv(data5, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data5.csv",row.names = FALSE)

write.csv(data6, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data6.csv",row.names = FALSE)

write.csv(data7, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data7.csv",row.names = FALSE)

write.csv(data8, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data8.csv",row.names = FALSE)

write.csv(data9, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data9.csv",row.names = FALSE)

write.csv(data10, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data10.csv",row.names = FALSE)

write.csv(data11, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data11.csv",row.names = FALSE)

write.csv(data12, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data12.csv",row.names = FALSE)

write.csv(data13, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data13.csv",row.names = FALSE)

write.csv(data14, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data14.csv",row.names = FALSE)

write.csv(data15, file = "C:\\Users\\35196\\Desktop\\Important_files\\Comparison_with_all\\Full_Data\\data15.csv",row.names = FALSE)

```


# Full data with GC and Splice Site scores 


```{r}
data2=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data2_withgc.csv")
data3=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data3_withgc.csv")
data4=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data4_withgc.csv")
data5=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data5_withgc.csv")
data6=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data6_withgc.csv")
data7=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data7_withgc.csv")
data8=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data8_withgc.csv")
data9=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data9_withgc.csv")
data10=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data10_withgc.csv")
data11=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data11_withgc.csv")
data12=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data12_withgc.csv")
data13=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data13_withgc.csv")
data14=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data14_withgc.csv")
#data15=read_csv("C:/Users/35196/Desktop/Important_files/Comparison_with_all/Full_Data/data15_withgc.csv")





splice_scores_IR <- read_table("C:/Users/35196/Desktop/Important_files/Matthieu-data/R_data/results/splice_scores_IR.txt")

data2= data2 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data3= data3 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data4= data4 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data5= data5 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data6= data6 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data7= data7 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data8= data8 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data9= data9 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data10= data10 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data11= data11 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data12= data12 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data13= data13 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

data14= data14 %>% 
  left_join (select(splice_scores_IR,Event_ID,'5_score','3_score'), by = c("EVENT" = "Event_ID"))

```


## Data with Estimate calculation



```{r}
data2 = data2 %>% 
  left_join(select(results2, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data3 = data3 %>% 
  left_join(select(results3, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data5 = data5 %>% 
  left_join(select(results5, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data4 = data4 %>% 
  left_join(select(results4, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data6 = data6 %>% 
  left_join(select(results6, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data8 = data8 %>% 
  left_join(select(results8, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data9 = data9 %>% 
  left_join(select(results9, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data10 = data10 %>% 
  left_join(select(results10, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data11 = data11 %>% 
  left_join(select(results11, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data12 = data12 %>% 
  left_join(select(results12, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

data14 = data14 %>% 
  left_join(select(results14, Estimate, StdError, ZValue, PValue, EventID), by = c("EVENT" = "EventID"))

```




















# Independent analyses

## Events per gene for Ribodepletion data

```{r}
events_per_gene_up <- data6 %>%
  filter(PSI_difference > 5) %>%
  count(Gene, name = "num_events")


events_pergene_up=ggplot(events_per_gene_up, aes(x = num_events)) +
  geom_histogram(binwidth = 0.5, fill = "#440154", color = "black") +
  theme_minimal() +
  labs(title = "Distribution of Events per Gene with deltaPSI more than 5",
       x = "Number of Events per Gene",
       y = "Number of Genes")


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_per_gene_ribo.png"), 
  
  # 2. Specify which plot object to save
  plot = events_pergene_up,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

events_per_gene_down <- data6 %>%
  filter(PSI_difference < -5) %>%
  count(Gene, name = "num_events")


events_pergene_down=ggplot(events_per_gene_down, aes(x = num_events)) +
  geom_histogram(binwidth = 0.5, fill = "#21908C", color = "black") +
  theme_minimal() +
  labs(title = "Distribution of Events per Gene with deltaPSI more than 5",
       x = "Number of Events per Gene",
       y = "Number of Genes")


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "events_per_gene_down_ribo.png"), 
  
  # 2. Specify which plot object to save
  plot = events_pergene_down,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```



# Plots


## Events per Gene

```{r}
bar_color <- "#4ECDC4"

# Create plot list
gene_barplots <- list()

library(ggplot2)
library(dplyr)

# Initialize a results dataframe
results <- data.frame(
  Dataset = 1:13,
  GeneCount = NA_integer_
)

# Count genes with events in each dataset
for (i in 1:13) {
  df <- data_list[[paste0("data", i)]]
  
  if (!is.null(df) && "GENE" %in% colnames(df)) {
    results$GeneCount[i] <- length(unique(df$GENE))
  } else {
    results$GeneCount[i] <- 0
  }
}

# Create the plot
ggplot(results, aes(x = factor(Dataset), y = GeneCount)) +
  geom_col(fill = "#4ECDC4", width = 0.7) +
  labs(
    title = "Genes with Splicing Events Across Datasets",
    x = "Dataset Number",
    y = "Number of Genes with ≥1 Event"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
    axis.text = element_text(size = 10),
    panel.grid.major.x = element_blank()
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
```
```{r}
library(ggplot2)
library(dplyr)


data_list <- list(data2, data3, data4, data5, data6, 
                  data7, data8, data9, data10, data11, data12, 
                  data13, data14)

# 1. Combine all datasets from data_list (assuming it contains all your data)
all_data <- bind_rows(data_list, .id = "Dataset")
library(ggplot2)
library(dplyr)

# 1. Count events per gene (compatible with all dplyr versions)
event_counts <- bind_rows(data_list, .id = "Dataset") %>% 
  group_by(GENE) %>% 
  summarise(Event_Count = n()) %>% 
  arrange(desc(Event_Count))

# 2. Plot top 20 genes
top_genes <- event_counts[1:20, ]

ggplot(top_genes, aes(x = reorder(GENE, Event_Count), y = Event_Count)) +
  geom_col(fill = "#1E88E5", width = 0.8) +
  geom_text(aes(label = Event_Count), 
            hjust = -0.2, 
            size = 3.5, 
            color = "black") +
  coord_flip() +
  labs(title = "Top 20 Genes by Splicing Events",
       x = "Gene",
       y = "Number of Events") +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
    axis.text.y = element_text(size = 11),
    panel.grid.major.y = element_blank()
  ) +
  expand_limits(y = max(top_genes$Event_Count) * 1.1) # Space for labels

# 3. Save plot
ggsave("gene_events.pdf", width = 9, height = 7, device = cairo_pdf)
```


```{r}
library(ggplot2)
library(dplyr)
library(patchwork)

# 1. Prepare data - count events per gene for each dataset
prepare_event_counts <- function(data_list, list_name) {
  bind_rows(lapply(names(data_list), function(dataset_name) {
    data_list[[dataset_name]] %>%
      count(Gene = GENE, name = "Count") %>% 
      mutate(Dataset = dataset_name,
             Group = list_name)
  }))
}

# Process both low and high ΔPSI groups
events_low <- prepare_event_counts(data_list_low, "Low ΔPSI")
events_high <- prepare_event_counts(data_list_high, "High ΔPSI")

# Combine all data
all_events <- bind_rows(events_low, events_high) %>%
  mutate(Dataset = factor(Dataset, levels = paste0("data", 1:14)),
         Group = factor(Group, levels = c("Low ΔPSI", "High ΔPSI")))

# 2. Create plotting function
plot_events_per_gene <- function(event_data, label_list) {
  ggplot(event_data, aes(x = Dataset, y = Count, fill = Group)) +
    geom_bar(stat = "identity", position = position_dodge(width = 0.8), width = 0.7) +
    scale_fill_manual(values = c("Low ΔPSI" = "#E69F00", "High ΔPSI" = "#56B4E9")) +
    scale_x_discrete(labels = label_list) +
    labs(title = "Splicing Events per Gene",
         x = "Dataset",
         y = "Number of Events",
         fill = "ΔPSI Group") +
    theme_minimal() +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
      plot.title = element_text(hjust = 0.5, face = "bold"),
      legend.position = "bottom"
    )
}

# 3. Generate plot
event_plot <- plot_events_per_gene(all_events, labels)

# 4. Save high-quality output
ggsave("events_per_gene.pdf", 
       plot = event_plot,
       device = cairo_pdf,
       width = 12,
       height = 6,
       units = "in")

# For interactive viewing
print(event_plot)
```






For the first analysis we will do a set of upset plots

First this upset plot represents the total number of events without any filter

```{r}
events_1 <- unique(data1_high$ID)
events_2 <- unique(data2_high$EVENT)
events_3 <- unique(data3_high$EVENT)
events_4 <- unique(data4_high$EVENT)
events_5 <- unique(data5_high$EVENT)
events_6 <- unique(data6_high$EVENT)
events_7 <- unique(data7_high$EVENT)
events_8 <- unique(data8_high$EVENT)
events_9 <- unique(data9_high$EVENT)
events_10 <- unique(data10_high$EVENT)
events_11 <- unique(data11_high$EVENT)
events_12 <- unique(data12_high$EVENT)
events_13 <- unique(data13_high$EVENT)
events_14 <- unique(data14_high$EVENT)


all_events <- unique(c(events_1,events_2, events_3, events_4, events_5, events_6, events_7,
                       events_8, events_9, events_10, events_11, events_12, events_13, events_14))


df_upset <- data.frame(
  `Proliferative vs Senescent BJ` = as.integer(all_events %in% events_1),
  `Proliferative vs Deep` = as.integer(all_events %in% events_2),
  `Proliferative vs Quiescent` = as.integer(all_events %in% events_3),
  `Proliferative vs Senescence` = as.integer(all_events %in% events_4),
  `Quiescent vs Senescence` = as.integer(all_events %in% events_7),
  `Senescence vs Deep` = as.integer(all_events %in% events_5),
  `Proliferative vs Senescence IMR ribodepletion` = as.integer(all_events %in% events_6)
)

rownames(df_upset) <- all_events

# Plot UpSet plot

upset(
  df_upset,
  nsets = ncol(df_upset),
  order.by = "freq",
  sets = colnames(df_upset),
  keep.order = TRUE,
  mb.ratio = c(0.65, 0.35),
  text.scale = c(1, 1, 1, 1, 1, 1),
  number.angles = 0,
  point.size = 2,
  line.size = 1.3,
  show.numbers = "yes"
)

```


## More than 5

```{r}
data1_high <- data1 %>% filter(PSI_difference > 5)
data2_high <- data2 %>% filter(PSI_difference > 5)
data3_high <- data3 %>% filter(PSI_difference > 5)
data4_high <- data4 %>% filter(PSI_difference > 5)
data5_high <- data5 %>% filter(PSI_difference > 5)
data6_high <- data6 %>% filter(PSI_difference > 5)
data7_high <- data7 %>% filter(PSI_difference > 5)
data8_high <- data8 %>% filter(PSI_difference > 5)
data9_high <- data9 %>% filter(PSI_difference > 5)
data10_high <- data10 %>% filter(PSI_difference > 5)
data11_high <- data11 %>% filter(PSI_difference > 5)
data12_high <- data12 %>% filter(PSI_difference > 5)
data13_high <- data13 %>% filter(PSI_difference > 5)
data14_high <- data14 %>% filter(PSI_difference > 5)





events_1 <- unique(data1_high$EventID)
events_2 <- unique(data2_high$EVENT)
events_3 <- unique(data3_high$EVENT)
events_4 <- unique(data4_high$EVENT)
events_5 <- unique(data5_high$EVENT)
events_6 <- unique(data6_high$EVENT)
events_7 <- unique(data7_high$EVENT)
events_8 <- unique(data8_high$EVENT)
events_9 <- unique(data9_high$EVENT)
events_10 <- unique(data10_high$EVENT)
events_11 <- unique(data11_high$EVENT)
events_12 <- unique(data12_high$EVENT)
events_13 <- unique(data13_high$EVENT)
events_14 <- unique(data14_high$EVENT)



all_events <- unique(c(events_1,events_2, events_3, events_4, events_5, events_6, events_7,
                       events_8, events_9, events_10, events_11, events_12, events_13, events_14))



df_upset <- data.frame(
  `Mattieu -- Proliferative vs Senescent` = as.integer(all_events %in% events_1),
  `Zhang -- Proliferative vs Deep` = as.integer(all_events %in% events_2),
  `Zhang -- Proliferative vs Quiescent` = as.integer(all_events %in% events_3),
  `Zhang -- Proliferative vs Senescence` = as.integer(all_events %in% events_4),
  `Zhang -- Quiescent vs Senescence` = as.integer(all_events %in% events_7),
  `Zhang -- Senescence vs Deep` = as.integer(all_events %in% events_5),
  `Casella -- Proliferative vs Senescence` = as.integer(all_events %in% events_6)
)
```



Now we filter only for datasets that have a senescence comparison


```{r}
senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))

library(ggplot2)

senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  sets = colnames(senescence_upset),
  keep.order = TRUE,
  mb.ratio = c(0.8, 0.2),
  text.scale = c(1, 1, 1, 1, 1, 1),
  number.angles = 0,
  point.size = 2,
  line.size = 1.3,
  show.numbers = "yes"
)


# Make sure the library is loaded
# library(UpSetR)

upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  
  # --- VISUAL & LAYOUT IMPROVEMENTS ---
  
  # 1. Balance the plot ratios. The default is too squished.
  mb.ratio = c(0.6, 0.4),
  
  # 2. Add clear labels to the axes.
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  
  # 3. Add a professional color scheme.
  main.bar.color = "steelblue",
  sets.bar.color = "skyblue",
  matrix.color = "midnightblue",
  shade.color = "#F0F0F0", # Light gray shading for matrix rows
  
  # 4. Refine text sizes for better hierarchy.
  # Order: intersection size title, intersection size tick labels, set size title, set size tick labels, set names, numbers on bars
  text.scale = c(1.5, 1.1, 1.5, 1.1, 1.2, 1),
  
  # 5. Let UpSetR reorder sets for a cleaner matrix view (often looks like a pyramid).
  keep.order = FALSE,
  
  # --- MINOR TWEAKS (from your original) ---
  number.angles = 0,
  point.size = 2.5, # Slightly larger points
  line.size = 1.3,
  show.numbers = "yes"
)


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 1.5, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)






```


This one is only for the zhang paper, because we want to compare also with deep senescence. So we selected the events that had deltaPSI > 5


```{r}
# Senescence
zhang_events = unique(c(events_2,events_4))

zhang_upset = data.frame(
  `Zhang BJ Proliferative vs Senescence` = as.integer(senescence_events %in% events_2),
  `Zhang BJ Proliferative vs Deep` = as.integer(senescence_events %in% events_4)
)


upset_plot=upset(
  zhang_upset,
  nsets = ncol(zhang_upset),
  order.by = "freq",
  sets = colnames(zhang_upset),
  keep.order = TRUE,
  mb.ratio = c(0.65, 0.35),
  text.scale = c(1, 1, 1, 1, 1, 1),
  number.angles = 0,
  point.size = 2,
  line.size = 1.3,
  show.numbers = "yes"
)

```


Now for zhang with quiescent 

```{r}
data3_high <- data3 %>% filter(PSI_difference > 5)
data4_high <- data4 %>% filter(PSI_difference > 5)

events_3 <- unique(data3_high$EVENT)
events_4 <- unique(data4_high$EVENT)


zhang_events = unique(c(events_3,events_4))

zhang_upset = data.frame(
  `Zhang BJ Proliferative vs Quiescent` = as.integer(senescence_events %in% events_3),
  `Zhang BJ Proliferative vs Senescence` = as.integer(senescence_events %in% events_4)
)


upset(
  zhang_upset,
  nsets = ncol(zhang_upset),
  order.by = "freq",
  sets = colnames(zhang_upset),
  keep.order = TRUE,
  mb.ratio = c(0.65, 0.35),
  text.scale = c(1, 1, 1, 1, 1, 1),
  number.angles = 0,
  point.size = 2,
  line.size = 1.3,
  show.numbers = "yes"
)
```


## Less than -5


```{r}
data1_low <- data1 %>% filter(PSI_difference < -5)
data2_low <- data2 %>% filter(PSI_difference < -5)
data3_low <- data3 %>% filter(PSI_difference < -5)
data4_low <- data4 %>% filter(PSI_difference < -5)
data5_low <- data5 %>% filter(PSI_difference < -5)
data6_low <- data6 %>% filter(PSI_difference < -5)
data7_low <- data7 %>% filter(PSI_difference < -5)
data8_low <- data8 %>% filter(PSI_difference < -5)
data9_low <- data9 %>% filter(PSI_difference < -5)
data10_low <- data10 %>% filter(PSI_difference < -5)
data11_low <- data11 %>% filter(PSI_difference < -5)
data12_low <- data12 %>% filter(PSI_difference < -5)
data13_low <- data13 %>% filter(PSI_difference < -5)
data14_low <- data14 %>% filter(PSI_difference < -5)


# Extract unique event IDs
events_1 <- unique(data1_low$EventID)
events_2 <- unique(data2_low$EVENT)
events_3 <- unique(data3_low$EVENT)
events_4 <- unique(data4_low$EVENT)
events_5 <- unique(data5_low$EVENT)
events_6 <- unique(data6_low$EVENT)
events_7 <- unique(data7_low$EVENT)
events_8 <- unique(data8_low$EVENT)
events_9 <- unique(data9_low$EVENT)
events_10 <- unique(data10_low$EVENT)
events_11 <- unique(data11_low$EVENT)
events_12 <- unique(data12_low$EVENT)
events_13 <- unique(data13_low$EVENT)
events_14 <- unique(data14_low$EVENT)



all_events <- unique(c(events_1,events_2, events_3, events_4, events_5, events_6, events_7,
                       events_8, events_9, events_10, events_11, events_12, events_13, events_14))

df_upset <- data.frame(
  `Mattieu -- Proliferative vs Senescent` = as.integer(all_events %in% events_1),
  `Zhang -- Proliferative vs Deep` = as.integer(all_events %in% events_2),
  `Zhang -- Proliferative vs Quiescent` = as.integer(all_events %in% events_3),
  `Zhang -- Proliferative vs Senescence` = as.integer(all_events %in% events_4),
  `Zhang -- Quiescent vs Senescence` = as.integer(all_events %in% events_7),
  `Zhang -- Senescence vs Deep` = as.integer(all_events %in% events_5),
  `Casella -- Proliferative vs Senescence` = as.integer(all_events %in% events_6)
)

rownames(df_upset) <- all_events

upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  sets = colnames(senescence_upset),
  keep.order = TRUE,
  mb.ratio = c(0.65, 0.35),
  text.scale = c(1, 1, 1, 1, 1, 1),
  number.angles = 0,
  point.size = 2,
  line.size = 1.3,
  show.numbers = "yes"
)
```


For Senescence events

```{r}
senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))


senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)

rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.4),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)

```


#Upsets with different cutoffs

2.5
 
```{r}
library(dplyr)
library(UpSetR)


data1_high_2 <- data1 %>% filter(PSI_difference > 2.5)
data2_high_2 <- data2 %>% filter(PSI_difference > 2.5)
data3_high_2 <- data3 %>% filter(PSI_difference > 2.5)
data4_high_2 <- data4 %>% filter(PSI_difference > 2.5)
data5_high_2 <- data5 %>% filter(PSI_difference > 2.5)
data6_high_2 <- data6 %>% filter(PSI_difference > 2.5)
data7_high_2 <- data7 %>% filter(PSI_difference > 2.5)
data8_high_2 <- data8 %>% filter(PSI_difference > 2.5)
data9_high_2 <- data9 %>% filter(PSI_difference > 2.5)
data10_high_2 <- data10 %>% filter(PSI_difference > 2.5)
data11_high_2 <- data11 %>% filter(PSI_difference > 2.5)
data12_high_2 <- data12 %>% filter(PSI_difference > 2.5)
data13_high_2 <- data13 %>% filter(PSI_difference > 2.5)
data14_high_2 <- data14 %>% filter(PSI_difference > 2.5)





events_1 <- unique(data1_high_2$EventID)
events_2 <- unique(data2_high_2$EVENT)
events_3 <- unique(data3_high_2$EVENT)
events_4 <- unique(data4_high_2$EVENT)
events_5 <- unique(data5_high_2$EVENT)
events_6 <- unique(data6_high_2$EVENT)
events_7 <- unique(data7_high_2$EVENT)
events_8 <- unique(data8_high_2$EVENT)
events_9 <- unique(data9_high_2$EVENT)
events_10 <- unique(data10_high_2$EVENT)
events_11 <- unique(data11_high_2$EVENT)
events_12 <- unique(data12_high_2$EVENT)
events_13 <- unique(data13_high_2$EVENT)
events_14 <- unique(data14_high_2$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)

```



-2.5

```{r}
library(dplyr)
library(UpSetR)


data1_low_2 <- data1 %>% filter(PSI_difference < -2.5)
data2_low_2 <- data2 %>% filter(PSI_difference < -2.5)
data3_low_2 <- data3 %>% filter(PSI_difference < -2.5)
data4_low_2 <- data4 %>% filter(PSI_difference < -2.5)
data5_low_2 <- data5 %>% filter(PSI_difference < -2.5)
data6_low_2 <- data6 %>% filter(PSI_difference < -2.5)
data7_low_2 <- data7 %>% filter(PSI_difference < -2.5)
data8_low_2 <- data8 %>% filter(PSI_difference < -2.5)
data9_low_2 <- data9 %>% filter(PSI_difference < -2.5)
data10_low_2 <- data10 %>% filter(PSI_difference < -2.5)
data11_low_2 <- data11 %>% filter(PSI_difference < -2.5)
data12_low_2 <- data12 %>% filter(PSI_difference < -2.5)
data13_low_2 <- data13 %>% filter(PSI_difference < -2.5)
data14_low_2 <- data14 %>% filter(PSI_difference < -2.5)





events_1 <- unique(data1_low_2$EventID)
events_2 <- unique(data2_low_2$EVENT)
events_3 <- unique(data3_low_2$EVENT)
events_4 <- unique(data4_low_2$EVENT)
events_5 <- unique(data5_low_2$EVENT)
events_6 <- unique(data6_low_2$EVENT)
events_7 <- unique(data7_low_2$EVENT)
events_8 <- unique(data8_low_2$EVENT)
events_9 <- unique(data9_low_2$EVENT)
events_10 <- unique(data10_low_2$EVENT)
events_11 <- unique(data11_low_2$EVENT)
events_12 <- unique(data12_low_2$EVENT)
events_13 <- unique(data13_low_2$EVENT)
events_14 <- unique(data14_low_2$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)

```


1

```{r}
library(dplyr)
library(UpSetR)


data1_high_3 <- data1 %>% filter(PSI_difference > 1)
data2_high_3 <- data2 %>% filter(PSI_difference > 1)
data3_high_3 <- data3 %>% filter(PSI_difference > 1)
data4_high_3 <- data4 %>% filter(PSI_difference > 1)
data5_high_3 <- data5 %>% filter(PSI_difference > 1)
data6_high_3 <- data6 %>% filter(PSI_difference > 1)
data7_high_3 <- data7 %>% filter(PSI_difference > 1)
data8_high_3 <- data8 %>% filter(PSI_difference > 1)
data9_high_3 <- data9 %>% filter(PSI_difference > 1)
data10_high_3 <- data10 %>% filter(PSI_difference > 1)
data11_high_3 <- data11 %>% filter(PSI_difference > 1)
data12_high_3 <- data12 %>% filter(PSI_difference > 1)
data13_high_3 <- data13 %>% filter(PSI_difference > 1)
data14_high_3 <- data14 %>% filter(PSI_difference > 1)





events_1 <- unique(data1_high_3$EventID)
events_2 <- unique(data2_high_3$EVENT)
events_3 <- unique(data3_high_3$EVENT)
events_4 <- unique(data4_high_3$EVENT)
events_5 <- unique(data5_high_3$EVENT)
events_6 <- unique(data6_high_3$EVENT)
events_7 <- unique(data7_high_3$EVENT)
events_8 <- unique(data8_high_3$EVENT)
events_9 <- unique(data9_high_3$EVENT)
events_10 <- unique(data10_high_3$EVENT)
events_11 <- unique(data11_high_3$EVENT)
events_12 <- unique(data12_high_3$EVENT)
events_13 <- unique(data13_high_3$EVENT)
events_14 <- unique(data14_high_3$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)
```

-1

```{r}
library(dplyr)
library(UpSetR)


data1_low_3 <- data1 %>% filter(PSI_difference < -1)
data2_low_3 <- data2 %>% filter(PSI_difference < -1)
data3_low_3 <- data3 %>% filter(PSI_difference < -1)
data4_low_3 <- data4 %>% filter(PSI_difference < -1)
data5_low_3 <- data5 %>% filter(PSI_difference < -1)
data6_low_3 <- data6 %>% filter(PSI_difference < -1)
data7_low_3 <- data7 %>% filter(PSI_difference < -1)
data8_low_3 <- data8 %>% filter(PSI_difference < -1)
data9_low_3 <- data9 %>% filter(PSI_difference < -1)
data10_low_3 <- data10 %>% filter(PSI_difference < -1)
data11_low_3 <- data11 %>% filter(PSI_difference < -1)
data12_low_3 <- data12 %>% filter(PSI_difference < -1)
data13_low_3 <- data13 %>% filter(PSI_difference < -1)
data14_low_3 <- data14 %>% filter(PSI_difference < -1)





events_1 <- unique(data1_low_3$EventID)
events_2 <- unique(data2_low_3$EVENT)
events_3 <- unique(data3_low_3$EVENT)
events_4 <- unique(data4_low_3$EVENT)
events_5 <- unique(data5_low_3$EVENT)
events_6 <- unique(data6_low_3$EVENT)
events_7 <- unique(data7_low_3$EVENT)
events_8 <- unique(data8_low_3$EVENT)
events_9 <- unique(data9_low_3$EVENT)
events_10 <- unique(data10_low_3$EVENT)
events_11 <- unique(data11_low_3$EVENT)
events_12 <- unique(data12_low_3$EVENT)
events_13 <- unique(data13_low_3$EVENT)
events_14 <- unique(data14_low_3$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)

```


0.5

```{r}
library(dplyr)
library(UpSetR)


data1_high_4 <- data1 %>% filter(PSI_difference > 0.5)
data2_high_4 <- data2 %>% filter(PSI_difference > 0.5)
data3_high_4 <- data3 %>% filter(PSI_difference > 0.5)
data4_high_4 <- data4 %>% filter(PSI_difference > 0.5)
data5_high_4 <- data5 %>% filter(PSI_difference > 0.5)
data6_high_4 <- data6 %>% filter(PSI_difference > 0.5)
data7_high_4 <- data7 %>% filter(PSI_difference > 0.5)
data8_high_4 <- data8 %>% filter(PSI_difference > 0.5)
data9_high_4 <- data9 %>% filter(PSI_difference > 0.5)
data10_high_4 <- data10 %>% filter(PSI_difference > 0.5)
data11_high_4 <- data11 %>% filter(PSI_difference > 0.5)
data12_high_4 <- data12 %>% filter(PSI_difference > 0.5)
data13_high_4 <- data13 %>% filter(PSI_difference > 0.5)
data14_high_4 <- data14 %>% filter(PSI_difference > 0.5)





events_1 <- unique(data1_high_4$EventID)
events_2 <- unique(data2_high_4$EVENT)
events_3 <- unique(data3_high_4$EVENT)
events_4 <- unique(data4_high_4$EVENT)
events_5 <- unique(data5_high_4$EVENT)
events_6 <- unique(data6_high_4$EVENT)
events_7 <- unique(data7_high_4$EVENT)
events_8 <- unique(data8_high_4$EVENT)
events_9 <- unique(data9_high_4$EVENT)
events_10 <- unique(data10_high_4$EVENT)
events_11 <- unique(data11_high_4$EVENT)
events_12 <- unique(data12_high_4$EVENT)
events_13 <- unique(data13_high_4$EVENT)
events_14 <- unique(data14_high_4$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)
```




-0.5


```{r}
library(dplyr)
library(UpSetR)


data1_low_4 <- data1 %>% filter(PSI_difference < -0.5)
data2_low_4 <- data2 %>% filter(PSI_difference < -0.5)
data3_low_4 <- data3 %>% filter(PSI_difference < -0.5)
data4_low_4 <- data4 %>% filter(PSI_difference < -0.5)
data5_low_4 <- data5 %>% filter(PSI_difference < -0.5)
data6_low_4 <- data6 %>% filter(PSI_difference < -0.5)
data7_low_4 <- data7 %>% filter(PSI_difference < -0.5)
data8_low_4 <- data8 %>% filter(PSI_difference < -0.5)
data9_low_4 <- data9 %>% filter(PSI_difference < -0.5)
data10_low_4 <- data10 %>% filter(PSI_difference < -0.5)
data11_low_4 <- data11 %>% filter(PSI_difference < -0.5)
data12_low_4 <- data12 %>% filter(PSI_difference < -0.5)
data13_low_4 <- data13 %>% filter(PSI_difference < -0.5)
data14_low_4 <- data14 %>% filter(PSI_difference < -0.5)





events_1 <- unique(data1_low_4$EventID)
events_2 <- unique(data2_low_4$EVENT)
events_3 <- unique(data3_low_4$EVENT)
events_4 <- unique(data4_low_4$EVENT)
events_5 <- unique(data5_low_4$EVENT)
events_6 <- unique(data6_low_4$EVENT)
events_7 <- unique(data7_low_4$EVENT)
events_8 <- unique(data8_low_4$EVENT)
events_9 <- unique(data9_low_4$EVENT)
events_10 <- unique(data10_low_4$EVENT)
events_11 <- unique(data11_low_4$EVENT)
events_12 <- unique(data12_low_4$EVENT)
events_13 <- unique(data13_low_4$EVENT)
events_14 <- unique(data14_low_4$EVENT)



senescence_events = unique(c(events_1, events_4, events_6, events_8, events_9, events_10, events_11, events_12, events_13, events_14))





senescence_upset <- data.frame(
  `Deschênes BJ` = as.integer(senescence_events %in% events_1),
  `Zhang BJ` = as.integer(senescence_events %in% events_4),
  `Casella IMR` = as.integer(senescence_events %in% events_6),
  `Marthandan BJ` = as.integer(senescence_events %in% events_8),
  `Marthandan HFF` = as.integer(senescence_events %in% events_9),
  `Marthandan IMR` = as.integer(senescence_events %in% events_10),
  `Marthandan WI-38` = as.integer(senescence_events %in% events_11),
  `Purcell MDAH041` = as.integer(senescence_events %in% events_12),
  `Purcell MDAH041 Pre` = as.integer(senescence_events %in% events_13),
  `Suda WI-38` = as.integer(senescence_events %in% events_14)
)


rownames(senescence_upset) <- senescence_events


upset(
  senescence_upset,
  nsets = ncol(senescence_upset),
  order.by = "freq",
  mb.ratio = c(0.6, 0.4),
  mainbar.y.label = "Intersection Size",
  sets.x.label = "Set Size",
  main.bar.color = "grey30",
  sets.bar.color = "skyblue",
  matrix.color = "grey10",
  shade.color = "#F0F0F0",
  text.scale = c(1.5, 1.3, 1.5, 1.4, 2, 1.6),
  keep.order = FALSE,
  number.angles = 0,
  point.size = 2.0,
  line.size = 1.1,
  show.numbers = "yes",
  scale.sets = "log10"  # <- compresses large set sizes
)

```






# Zhang comparions


here I will compare the PSI across conditions




Now I will do the same but with Heatmaps

This one is for Data 4 

```{r}
#-----------------------------------------------------------------------
# Step 1: Load Libraries
#-----------------------------------------------------------------------
# Make sure you have these libraries loaded
library(tidyverse)
library(pheatmap)


#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data4_filtered = data4 %>%
  filter(PSI_difference>5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data4_filtered$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data4_filtered[proliferative_samples]),
  Quiescence      = rowMeans(data4_filtered[quiescence_samples]),
  Senescence      = rowMeans(data4_filtered[senescence_samples]),
  Deep_Senescence = rowMeans(data4_filtered[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data4"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data4.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```

Now for  data 2

```{r}


data2_filtered = data2 %>%
  filter(PSI_difference>5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data2_filtered$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data2_filtered[proliferative_samples]),
  Quiescence      = rowMeans(data2_filtered[quiescence_samples]),
  Senescence      = rowMeans(data2_filtered[senescence_samples]),
  Deep_Senescence = rowMeans(data2_filtered[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data2"
)

# View the plot
print(heatmap_new_colors)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data2.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```










Now for data 3













```{r}

data3_filtered = data3 %>%
  filter(PSI_difference>5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data3_filtered$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data3_filtered[proliferative_samples]),
  Quiescence      = rowMeans(data3_filtered[quiescence_samples]),
  Senescence      = rowMeans(data3_filtered[senescence_samples]),
  Deep_Senescence = rowMeans(data3_filtered[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data3"
)

# View the plot
print(heatmap_new_colors)

ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data3.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```




Now for data 7




```{r}

data7_filtered = data7 %>%
  filter(PSI_difference>5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data7_filtered$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data7_filtered[proliferative_samples]),
  Quiescence      = rowMeans(data7_filtered[quiescence_samples]),
  Senescence      = rowMeans(data7_filtered[senescence_samples]),
  Deep_Senescence = rowMeans(data7_filtered[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data7"
)

# View the plot
print(heatmap_new_colors)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data7.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```


For Data5

```{r}

data5_filtered = data5 %>%
  filter(PSI_difference>5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data5_filtered$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data5_filtered[proliferative_samples]),
  Quiescence      = rowMeans(data5_filtered[quiescence_samples]),
  Senescence      = rowMeans(data5_filtered[senescence_samples]),
  Deep_Senescence = rowMeans(data5_filtered[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data5"
)

# View the plot
print(heatmap_new_colors)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data5.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```

## Heatmaps for deltaPSI < -5



data4

```{r}

#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data4_filtered_less = data4 %>%
  filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data4_filtered_less$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data4_filtered_less[proliferative_samples]),
  Quiescence      = rowMeans(data4_filtered_less[quiescence_samples]),
  Senescence      = rowMeans(data4_filtered_less[senescence_samples]),
  Deep_Senescence = rowMeans(data4_filtered_less[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data4 with deltaPSI < -5"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data4_less.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```
Data5

```{r}

#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data5_filtered_less = data5 %>%
  filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data5_filtered_less$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data5_filtered_less[proliferative_samples]),
  Quiescence      = rowMeans(data5_filtered_less[quiescence_samples]),
  Senescence      = rowMeans(data5_filtered_less[senescence_samples]),
  Deep_Senescence = rowMeans(data5_filtered_less[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data5 with deltaPSI < -5"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data5_less.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```
Data7

```{r}

#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data7_filtered_less = data7 %>%
  filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data7_filtered_less$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data7_filtered_less[proliferative_samples]),
  Quiescence      = rowMeans(data7_filtered_less[quiescence_samples]),
  Senescence      = rowMeans(data7_filtered_less[senescence_samples]),
  Deep_Senescence = rowMeans(data7_filtered_less[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data7 with deltaPSI < -5"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data7_less.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```

```{r}

#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data2_filtered_less = data2 %>%
  filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data2_filtered_less$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data2_filtered_less[proliferative_samples]),
  Quiescence      = rowMeans(data2_filtered_less[quiescence_samples]),
  Senescence      = rowMeans(data2_filtered_less[senescence_samples]),
  Deep_Senescence = rowMeans(data2_filtered_less[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data2 with deltaPSI < -5"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data2_less.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```



```{r}

#-----------------------------------------------------------------------
# Step 2: Define your sample groups
#-----------------------------------------------------------------------
# This list maps your sample columns to their biological condition.
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")



data3_filtered_less = data3 %>%
  filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Step 3: Aggregate your `data4` dataframe by condition
#-----------------------------------------------------------------------
# This calculates the mean PSI for each group of samples for every event.
# It now correctly uses the "EventID" column for the identifiers.
psi_aggregated <- tibble(
  # --- THIS IS THE UPDATED LINE ---
  # We get the event names directly from the EventID column in your data.
  event_id = data3_filtered_less$EVENT, 
  
  # The rowMeans calculations remain the same.
  Proliferative   = rowMeans(data3_filtered_less[proliferative_samples]),
  Quiescence      = rowMeans(data3_filtered_less[quiescence_samples]),
  Senescence      = rowMeans(data3_filtered_less[senescence_samples]),
  Deep_Senescence = rowMeans(data3_filtered_less[deep_senescence_samples])
)


#-----------------------------------------------------------------------
# Step 4: Prepare the data for the heatmap function
#-----------------------------------------------------------------------
# Convert the aggregated data into a matrix format with event IDs as row names,
# which is what pheatmap requires.
psi_matrix <- psi_aggregated %>%
  column_to_rownames("event_id") %>%
  as.matrix()



#-----------------------------------------------------------------------
# Step 5 (Revised): Heatmap with a new Color Palette
#-----------------------------------------------------------------------

# --- NEW COLOR PALETTE: Classic Yellow-Orange-Red ---
# This palette provides excellent contrast and is very intuitive for "hot" (high)
# and "cold" (low) values.
# We use colorRampPalette to create a smooth gradient of 100 colors.
my_colors_sequential <- colorRampPalette(RColorBrewer::brewer.pal(n = 9, name = "YlOrRd"))(100)


# --- The rest of your code remains exactly the same ---

# The non-linear breaks are still very important!
breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)

# Generate the heatmap with the new colors
heatmap_new_colors <- pheatmap(
  psi_matrix,
  scale = "none",
  color = my_colors_sequential, # The new palette is used here
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Data3 with deltaPSI < -5"
)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_data3_less.png"), 
  
  # 2. Specify which plot object to save
  plot = heatmap_new_colors,
  
  # 3. Set the dimensions and resolution
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)
```

## Common events for Zhang ALLL WITH DATA15

```{r}
# Make sure all EVENT columns are character and trimmed
events2 <- unique(trimws(as.character(data2$EVENT)))
events3 <- unique(trimws(as.character(data3$EVENT)))
events4 <- unique(trimws(as.character(data4$EVENT)))
events5 <- unique(trimws(as.character(data5$EVENT)))
events7 <- unique(trimws(as.character(data7$EVENT)))
events15 <- unique(trimws(as.character(data15$EVENT)))

data15_filtered = data15 %>%
  filter(PSI_difference>5)



# Get the intersection across all 5 datasets
common_eventIDs_all <- Reduce(intersect, list(events2, events3, events4, events5, events7, events15))

# See how many common events exist
length(common_eventIDs_all)


# Step 1: Filter each filtered dataset to include only common events
data2_common <- data2_filtered %>% filter(EVENT %in% common_eventIDs_all)
data3_common <- data3_filtered %>% filter(EVENT %in% common_eventIDs_all)
data4_common <- data4_filtered %>% filter(EVENT %in% common_eventIDs_all)
data5_common <- data5_filtered %>% filter(EVENT %in% common_eventIDs_all)
data7_common <- data7_filtered %>% filter(EVENT %in% common_eventIDs_all)
data15_common <- data15_filtered %>% filter(EVENT %in% common_eventIDs_all)



# Step 2: Combine all into one global dataset
global_data_up <- bind_rows(
  data2_common,
  data3_common,
  data4_common,
  data5_common,
  data7_common,
  data15_common
)


write.csv(global_data, "C:\\Users\\35196\\Desktop\\Important_files\\global_data.csv", row.names = FALSE)
```




```{r}
#-----------------------------------------------------------------------
# Define your sample groups
#-----------------------------------------------------------------------
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")


#-----------------------------------------------------------------------
# OPTIONAL: Filter by PSI_difference < -5 if desired
#-----------------------------------------------------------------------
# global_data <- global_data %>% filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Aggregate the PSI by condition
#-----------------------------------------------------------------------
psi_aggregated_global <- tibble(
  event_id = global_data_up$EVENT,
  
  Proliferative   = rowMeans(global_data_up[proliferative_samples], na.rm = TRUE),
  Quiescence      = rowMeans(global_data_up[quiescence_samples], na.rm = TRUE),
  Senescence      = rowMeans(global_data_up[senescence_samples], na.rm = TRUE),
  Deep_Senescence = rowMeans(global_data_up[deep_senescence_samples], na.rm = TRUE)
)

# Optional: remove duplicated event IDs
psi_aggregated_global <- psi_aggregated_global %>%
  distinct(event_id, .keep_all = TRUE)

#
#-----------------------------------------------------------------------
# Prepare the matrix for heatmap
#-----------------------------------------------------------------------
psi_matrix_global <- psi_aggregated_global %>%
  column_to_rownames("event_id") %>%
  as.matrix()


#-----------------------------------------------------------------------
# Define color palette and breaks
#-----------------------------------------------------------------------
my_colors_sequential <- colorRampPalette(
  RColorBrewer::brewer.pal(n = 9, name = "YlOrRd")
)(100)

breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)


#-----------------------------------------------------------------------
# Generate the heatmap
#-----------------------------------------------------------------------
heatmap_global <- pheatmap(
  psi_matrix_global,
  scale = "none",
  color = my_colors_sequential,
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Common Events Across All Data Sets"
)


library(dendextend)  # optional for nicer dendrogram handling



# Assuming you already have the clusters vector from cutree:
clusters <- cutree(heatmap_global$tree_row, k = 10)

# Create a data frame with event IDs and their cluster assignment
cluster_table <- data.frame(
  event_id = names(clusters),
  cluster = clusters,
  row.names = NULL
)

# View the first few rows
head(cluster_table)


write.csv(cluster_table, "C:\\Users\\35196\\Desktop\\Important_files\\cluster_table.csv", row.names = FALSE)


#-----heatmap_global#-----------------------------------------------------------------------
# Save the heatmap
#-----------------------------------------------------------------------
ggsave(
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_global_common_events_withdata15.png"),
  plot = heatmap_global,
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,
  type = "cairo"
)

```








```{r}
# Make sure all EVENT columns are character and trimmed
events2 <- unique(trimws(as.character(data2$EVENT)))
events3 <- unique(trimws(as.character(data3$EVENT)))
events4 <- unique(trimws(as.character(data4$EVENT)))
events5 <- unique(trimws(as.character(data5$EVENT)))
events7 <- unique(trimws(as.character(data7$EVENT)))
events15 <- unique(trimws(as.character(data15$EVENT)))


data15_filtered_less = data15 %>%
  filter(PSI_difference < -5)




# Get the intersection across all 5 datasets
common_eventIDs_all <- Reduce(intersect, list(events2, events3, events4, events5, events7, events15))

# See how many common events exist
length(common_eventIDs_all)


# Step 1: Filter each filtered dataset to include only common events
data2_common <- data2_filtered_less %>% filter(EVENT %in% common_eventIDs_all)
data3_common <- data3_filtered_less %>% filter(EVENT %in% common_eventIDs_all)
data4_common <- data4_filtered_less %>% filter(EVENT %in% common_eventIDs_all)
data5_common <- data5_filtered_less %>% filter(EVENT %in% common_eventIDs_all)
data7_common <- data7_filtered_less %>% filter(EVENT %in% common_eventIDs_all)
data15_common <- data15_filtered_less %>% filter(EVENT %in% common_eventIDs_all)



# Step 2: Combine all into one global dataset
global_data_down <- bind_rows(
  data2_common,
  data3_common,
  data4_common,
  data5_common,
  data7_common,
  data15_common
)


#write.csv(global_data, "C:\\Users\\35196\\Desktop\\Important_files\\global_data.csv", row.names = FALSE)
```




```{r}
#-----------------------------------------------------------------------
# Define your sample groups
#-----------------------------------------------------------------------
proliferative_samples <- c("SRR9601022", "SRR9601023")
quiescence_samples <- c("SRR9601024", "SRR9601025")
senescence_samples <- c("SRR9601026", "SRR9601027")
deep_senescence_samples <- c("SRR9601028", "SRR9601029")


#-----------------------------------------------------------------------
# OPTIONAL: Filter by PSI_difference < -5 if desired
#-----------------------------------------------------------------------
# global_data <- global_data %>% filter(PSI_difference < -5)


#-----------------------------------------------------------------------
# Aggregate the PSI by condition
#-----------------------------------------------------------------------
#-----------------------------------------------------------------------
psi_aggregated_global <- tibble(
  event_id = global_data_down$EVENT,
  
  Proliferative   = rowMeans(global_data_down[proliferative_samples], na.rm = TRUE),
  Quiescence      = rowMeans(global_data_down[quiescence_samples], na.rm = TRUE),
  Senescence      = rowMeans(global_data_down[senescence_samples], na.rm = TRUE),
  Deep_Senescence = rowMeans(global_data_down[deep_senescence_samples], na.rm = TRUE)
)

# Optional: remove duplicated event IDs
psi_aggregated_global <- psi_aggregated_global %>%
  distinct(event_id, .keep_all = TRUE)




psi_matrix_global_less <- psi_aggregated_global %>%
  column_to_rownames("event_id") %>%
  as.matrix()


#-----------------------------------------------------------------------
# Define color palette and breaks
#-----------------------------------------------------------------------
my_colors_sequential <- colorRampPalette(
  RColorBrewer::brewer.pal(n = 9, name = "YlOrRd")
)(100)

breaks_low <- seq(0, 30, length.out = 70) 
breaks_high <- seq(30.01, 100, length.out = 31)
my_breaks_nonlinear <- c(breaks_low, breaks_high)


#-----------------------------------------------------------------------
# Generate the heatmap
#-----------------------------------------------------------------------
heatmap_global_less <- pheatmap(
  psi_matrix_global,
  scale = "none",
  color = my_colors_sequential,
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for Common Events Across All Data Sets with deltaPSI < -5"
)


library(dendextend)  # optional for nicer dendrogram handling



# Assuming you already have the clusters vector from cutree:
clusters <- cutree(heatmap_global$tree_row, k = 10)

# Create a data frame with event IDs and their cluster assignment
cluster_table <- data.frame(
  event_id = names(clusters),
  cluster = clusters,
  row.names = NULL
)

# View the first few rows
head(cluster_table)


#(cluster_table, "C:\\Users\\35196\\Desktop\\Important_files\\cluster_table.csv", row.names = FALSE)


#-----heatmap_global#-----------------------------------------------------------------------
# Save the heatmap
#-----------------------------------------------------------------------
ggsave(
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_global_common_events_less_withdata15.png"),
  plot = heatmap_global_less,
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,
  type = "cairo"
)

```




### Combine


```{r}
# You already have these matrices from each run:
# psi_matrix_global       # for ΔPSI > 5
# psi_matrix_global_less  # for ΔPSI < -5

# 1. Combine the two matrices by row-binding
combined_matrix <- rbind(
  psi_matrix_global,
  psi_matrix_global_less
)

# 2. Generate one heatmap from the combined matrix
heatmap_combined <- pheatmap(
  combined_matrix,
  scale = "none",
  color = my_colors_sequential,
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for ΔPSI > 5 and ΔPSI < -5"
)


# Combine the matrices
combined_matrix <- rbind(
  psi_matrix_global,
  psi_matrix_global_less
)

# Remove duplicated rows based on rownames (event_id)
combined_matrix <- combined_matrix[!duplicated(rownames(combined_matrix)), ]

# Check that duplicates are gone
any(duplicated(rownames(combined_matrix)))  # should return FALSE

# Generate the heatmap
heatmap_combined <- pheatmap(
  combined_matrix,
  scale = "none",
  color = my_colors_sequential,
  breaks = my_breaks_nonlinear,
  legend_title = "Mean PSI (%)",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  angle_col = 0,
  fontsize = 12,
  fontsize_col = 10,
  border_color = NA,
  main = "PSI Distribution for ΔPSI > 5 and ΔPSI < -5"
)

ggsave(
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "heatmap_unified_with_data15.png"),
  plot = heatmap_combined,
  width = 10,
  height = 7,
  units = "in",
  dpi = 300,
  type = "cairo"
)


```





# Intron Length

For deltaPSI > 5



2nd way of doing the histograms





```{r}
labels <- c(
  "Mattieu BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
  "Purcell MDAH041", "Zhang BJ", "Zhang BJ",
  "Zhang BJ", "Zhang BJ"
)

add_psi_category <- function(df) {
  df %>%
    mutate(
      psi_category = case_when(
        PSI_difference > 5 ~ "PSI > 5",
        PSI_difference < -5 ~ "PSI < -5",
        between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
        TRUE ~ "Other"
      ),
      psi_category = factor(psi_category, levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other"))
    )
}

data_list <- list(data1,data4,data6,data8,data9,data10,data11,
                  data12,data14,data13,data3,data7,data2,data5)
names(data_list) <- paste0("data", 1:14)
data_list_categorized <- lapply(data_list, add_psi_category)

plot_psi_comparison <- function(df, title_text, label_text) {
  df_filtered <- df %>% filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05"))
  
  ggplot(df_filtered, aes(x = log(LENGTH + 1), fill = psi_category)) +
    geom_histogram(
      position = "dodge",
      alpha = 1,
      bins = 20,
      color = "black",
      linewidth = 0.3
    ) +
    scale_fill_manual(
      values = c(
        "PSI > 5" = "#440154",
        "PSI < -5" = "#21908C",
        "-0.05 < PSI < 0.05" = "grey70"
      ),
      name = "PSI Difference"
    ) +
    theme_minimal() +
    labs(
      title = title_text,      # updated to use custom title
      subtitle = label_text,   # keep the same subtitle
      x = "Log LENGTH",
      y = "Count"
    ) +
    theme(
      plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
      plot.subtitle = element_text(hjust = 0.5, face = "italic", size = 10),
      axis.title = element_text(face = "bold"),
      legend.position = "bottom"
    )
}

# New titles for each dataset in order
new_titles <- c(
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Pre-Senescent vs Proliferative",
  "Quiescent vs Proliferative",
  "Senescent vs Quiescent",
  "Deep-Senescent vs Proliferative",
  "Deep-Senescent vs Senescent"
)

# Create plots with updated titles
plots <- mapply(function(nm, lbl, ttl) {
  plot_psi_comparison(data_list_categorized[[nm]], ttl, lbl)
}, names(data_list_categorized), labels, new_titles, SIMPLIFY = FALSE)

# Combine plots as before
length_plot = wrap_plots(plots, ncol = 4) + 
  plot_annotation(
    title = "Intron Length Distribution by PSI Difference",
    theme = theme(plot.title = element_text(size = 20, face = "bold", hjust = 0.5))
  ) +
  plot_layout(guides = "collect") & theme(legend.position = 'bottom')

print(length_plot)


ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "length_comparison.png"), 
  
  # 2. Specify which plot object to save
  plot = length_plot,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 7,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```










### STATISTICAL FOR INTRON



```{r}
library(dplyr)
library(tidyr)

# --- Data list in the same custom order you use for plotting ---
data_list <- list(
  data1, data4, data6, data8, data9, data10, data11,
  data12, data14, data13, data3, data7, data2, data5
)

# Give them names exactly as you want in the CSV (data1, data4, etc.)
names(data_list) <- c(
  "data1", "data4", "data6", "data8", "data9", "data10", "data11",
  "data12", "data14", "data13", "data3", "data7", "data2", "data5"
)

# --- Function to run stats ---
extract_stats <- function(df, dataset_name) {
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference <= 0.05 & PSI_difference >= -0.05 ~ "Near Zero (|ΔPSI| < 0.05)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(LENGTH)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.05)")
    ))
  
  # Kruskal-Wallis
  kruskal_res <- kruskal.test(log(LENGTH) ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$LENGTH,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Pairwise tidy
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # Medians
  medians <- plot_data_length %>%
    group_by(psi_group) %>%
    summarise(median_length = median(LENGTH, na.rm = TRUE), .groups = "drop")
  
  # Assemble tidy output
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_length) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_length) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Loop preserving the custom order ---
all_stats <- lapply(names(data_list), function(nm) {
  extract_stats(data_list[[nm]], nm)
})

# Combine results into one dataframe
all_stats_df <- bind_rows(all_stats)

# --- Write to CSV ---
write.csv(all_stats_df, "C:/Users/35196/Desktop/all_datasets_stat_tests.csv", row.names = FALSE)

# Quick check
head(all_stats_df)

```




# Integration of statistical with Plots


```{r}
# --- Libraries ---
library(dplyr)
library(tidyr)
library(ggplot2)
library(patchwork)
library(tibble)

# --- Labels for subtitles (each dataset) ---
labels <- c(
  "Deschênes BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
  "Purcell MDAH041", "Zhang BJ", "Zhang BJ",
  "Zhang BJ", "Zhang BJ"
)

# --- Add PSI categories to each dataset ---
add_psi_category <- function(df) {
  df %>%
    mutate(
      psi_category = case_when(
        PSI_difference > 5 ~ "PSI > 5",
        PSI_difference < -5 ~ "PSI < -5",
        between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
        TRUE ~ "Other"
      ),
      psi_category = factor(psi_category,
                            levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other"))
    )
}

# --- Collect all datasets ---
data_list <- list(data1,data4,data6,data8,data9,data10,data11,
                  data12,data14,data13,data3,data7,data2,data5)
names(data_list) <- paste0("data", 1:14)
data_list_categorized <- lapply(data_list, add_psi_category)

# --- Titles for each dataset (main plot title) ---
new_titles <- c(
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Pre-Senescent vs Proliferative",
  "Quiescent vs Proliferative",
  "Senescent vs Quiescent",
  "Deep-Senescent vs Proliferative",
  "Deep-Senescent vs Senescent"
)

# ============================================================
# --- Statistical tests function ---
# ============================================================
extract_stats <- function(df, dataset_name) {
  plot_data_length <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                            ~ "High ΔPSI (> 5)",
      PSI_difference < -5                           ~ "Low ΔPSI (< -5)",
      PSI_difference <= 0.05 & PSI_difference >= -0.1  ~ "Near Zero (|ΔPSI| < 0.05)",
      TRUE                                          ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(LENGTH)) %>%
    mutate(psi_group = factor(psi_group,
      levels = c("High ΔPSI (> 5)",
                 "Low ΔPSI (< -5)",
                 "Near Zero (|ΔPSI| < 0.05)")
    ))
  
  # Kruskal-Wallis
  kruskal_res <- kruskal.test(log(LENGTH) ~ psi_group, data = plot_data_length)
  
  # Pairwise Wilcoxon
  pairwise_res <- pairwise.wilcox.test(
    plot_data_length$LENGTH,
    plot_data_length$psi_group,
    p.adjust.method = "none"
  )
  
  # Tidy pairwise results
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value)) %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    select(dataset, kruskal_statistic, kruskal_p_value, group1, group2, p_value)
  
  return(pairwise_df)
}

# Run stats for all datasets
all_stats <- mapply(
  extract_stats,
  df = data_list,
  dataset_name = names(data_list),
  SIMPLIFY = FALSE
)
all_stats_df <- bind_rows(all_stats)

# ============================================================
# --- Map comparisons to letters (A, B, C) ---
# ============================================================
# Define mapping in both directions
comparison_map <- tribble(
  ~group1,                   ~group2,                    ~letter,
  "High ΔPSI (> 5)",         "Low ΔPSI (< -5)",          "A",
  "Low ΔPSI (< -5)",         "High ΔPSI (> 5)",          "A",
  "High ΔPSI (> 5)",         "Near Zero (|ΔPSI| < 0.05)", "B",
  "Near Zero (|ΔPSI| < 0.05)","High ΔPSI (> 5)",          "B",
  "Low ΔPSI (< -5)",         "Near Zero (|ΔPSI| < 0.05)", "C",
  "Near Zero (|ΔPSI| < 0.05)","Low ΔPSI (< -5)",          "C"
)


all_stats_df_labeled <- all_stats_df %>%
  left_join(comparison_map, by = c("group1", "group2")) %>%
  mutate(sig_label = ifelse(p_value < 0.05, letter, NA))

dataset_labels <- all_stats_df_labeled %>%
  group_by(dataset) %>%
  summarize(
    annotations = paste(na.omit(sig_label), collapse = ", ")
  )


# ============================================================
# --- Plotting functions ---
# ============================================================

# 1) Original (no annotations)
plot_psi_comparison <- function(df, title_text, label_text) {
  df_filtered <- df %>% filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05"))
  
  ggplot(df_filtered, aes(x = log(LENGTH), fill = psi_category)) +
    geom_histogram(
      position = "dodge",
      alpha = 1,
      bins = 20,
      color = "black",
      linewidth = 0.3
    ) +
    scale_fill_manual(
      values = c(
        "PSI > 5" = "#440154",
        "PSI < -5" = "#21908C",
        "-0.05 < PSI < 0.05" = "grey70"
      ),
      name = "PSI Difference"
    ) +
    theme_minimal() +
    labs(
      title = title_text,
      subtitle = label_text,
      x = "Log LENGTH",
      y = "Count"
    ) +
    theme(
      plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
      plot.subtitle = element_text(hjust = 0.5, face = "italic", size = 10),
      axis.title = element_text(face = "bold"),
      legend.position = "bottom"
    )
}

# 2) Annotated (adds letters A, B, C if significant)
plot_psi_comparison_annotated <- function(df, title_text, label_text, dataset_name, dataset_labels) {
  df_filtered <- df %>% filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05"))
  
  # Base plot
  p <- ggplot(df_filtered, aes(x = log(LENGTH), fill = psi_category)) +
    geom_histogram(
      position = "dodge",
      alpha = 1,
      bins = 20,
      color = "black",
      linewidth = 0.3
    ) +
    scale_fill_manual(
      values = c(
        "PSI > 5" = "#440154",
        "PSI < -5" = "#21908C",
        "-0.05 < PSI < 0.05" = "grey70"
      ),
      name = "PSI Difference"
    ) +
    theme_minimal() +
    labs(
      title = title_text,
      subtitle = label_text,
      x = "Log LENGTH",
      y = "Count"
    ) +
    theme(
      plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
      plot.subtitle = element_text(hjust = 0.5, face = "italic", size = 10),
      #axis.title = element_text(face = "bold"),
      legend.position = "bottom"
    )
  
  # Add annotation if present
  annotation_text <- dataset_labels %>%
    filter(dataset == dataset_name) %>%
    pull(annotations)
  
  if (length(annotation_text) > 0 && annotation_text != "") {
    p <- p + annotate(
      "text",
      x = Inf, y = Inf,
      label = annotation_text,
      hjust = 1.1, vjust = 1.5,
      size = 5, fontface = "bold"
    )
  }
  
  return(p)
}

# ============================================================
# --- Generate plots ---
# ============================================================

# Original plots
plots_clean <- mapply(function(nm, lbl, ttl) {
  plot_psi_comparison(data_list_categorized[[nm]], ttl, lbl)
}, names(data_list_categorized), labels, new_titles, SIMPLIFY = FALSE)

length_plot_clean <- wrap_plots(plots_clean, ncol = 4) +
  plot_annotation(
    title = "Intron Length Distribution by PSI Difference",
    theme = theme(plot.title = element_text(size = 20, face = "bold", hjust = 0.5))
  ) +
  plot_layout(guides = "collect") & theme(legend.position = 'bottom')

# Annotated plots
plots_annotated <- mapply(function(nm, lbl, ttl) {
  plot_psi_comparison_annotated(data_list_categorized[[nm]], ttl, lbl, nm, dataset_labels)
}, names(data_list_categorized), labels, new_titles, SIMPLIFY = FALSE)

length_plot_annotated <- wrap_plots(plots_annotated, ncol = 4) +
  plot_annotation(
    title = "Intron Length Distribution by PSI Difference (with significance letters)",
    theme = theme(plot.title = element_text(size = 20, face = "bold", hjust = 0.5))
  ) +
  plot_layout(guides = "collect") & theme(legend.position = 'bottom')


# ============================================================
# --- Print the results ---
# ============================================================
#print(length_plot_clean)      # Original 14-panel grid
#print(length_plot_annotated)  # Annotated version with A/B/C

```



```{r}
library(patchwork)
library(ggplot2)

# --- Top 9 plots ---
plots_top <- mapply(function(nm, lbl) {
  plot_psi_comparison_annotated(
    df = data_list_categorized[[nm]],
    title_text = "",          
    label_text = lbl,
    dataset_name = nm,
    dataset_labels = dataset_labels
  )
}, names(data_list_categorized)[1:9], labels[1:9], SIMPLIFY = FALSE)

# --- Fake title as a ggplot object ---
top_title <- ggplot() +
  annotate("text", x = 0.5, y = 0.5, label = "Senescence vs Proliferative", size = 6, fontface = "bold") +
  theme_void()

# --- Bottom 5 plots ---
plots_bottom <- mapply(function(nm, lbl, ttl) {
  plot_psi_comparison_annotated(
    df = data_list_categorized[[nm]],
    title_text = ttl,          
    label_text = lbl,
    dataset_name = nm,
    dataset_labels = dataset_labels
  )
}, names(data_list_categorized)[10:14], labels[10:14], new_titles[10:14], SIMPLIFY = FALSE)

# --- Fill bottom to make 3x3 ---
num_spacers <- 9 - length(plots_bottom)
plots_bottom <- c(plots_bottom, rep(list(plot_spacer()), num_spacers))

# --- Combine grids ---
top_grid <- wrap_plots(plots_top, ncol = 3)
bottom_grid <- wrap_plots(plots_bottom, ncol = 3)

# --- Optional dashed separator ---
separator <- ggplot() + geom_hline(yintercept = 0, linetype = "dashed", color = "grey50") + theme_void()

# --- Stack everything with the fake title on top ---
final_plot <- top_title / top_grid / separator / bottom_grid +
  plot_layout(heights = c(0.2, 3, 0.05, 2), guides = "collect") &
  theme(legend.position = 'bottom')

# --- Print final plot ---
print(final_plot)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "length_all.png"), 
  
  # 2. Specify which plot object to save
  plot = final_plot,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 12,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```



# GC content



```{r}
library(dplyr)
library(ggplot2)
library(patchwork)
library(ggpubr)

# -----------------------------
# 1️⃣ Define PSI categorization
# -----------------------------
add_psi_category <- function(df) {
  df %>%
    mutate(
      psi_category = case_when(
        PSI_difference > 5  ~ "PSI > 5",
        PSI_difference < -5 ~ "PSI < -5",
        between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
        TRUE ~ "Other"
      ),
      psi_category = factor(
        psi_category,
        levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other")
      )
    )
}

# -----------------------------
# 2️⃣ Prepare GC content data
# -----------------------------
prepare_gc_data <- function(df) {
  df %>%
    add_psi_category() %>%
    filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05")) %>%
    mutate(Group = case_when(
      psi_category == "PSI > 5"  ~ "High ΔPSI",
      psi_category == "PSI < -5" ~ "Low ΔPSI",
      psi_category == "-0.05 < PSI < 0.05" ~ "No ΔPSI"
    ),
    Group = factor(Group, levels = c("Low ΔPSI", "No ΔPSI", "High ΔPSI")))
}

# -----------------------------
# 3️⃣ Create violin + boxplot
# -----------------------------
create_gc_plot <- function(df, title, subtitle) {
  plot_data <- prepare_gc_data(df) %>% filter(!is.na(GC_Ratio))
  
  if(nrow(plot_data) == 0){
    return(ggplot() + labs(title = title, subtitle = "No data") + theme_void())
  }
  
  y_max <- max(plot_data$GC_Ratio, na.rm = TRUE)
  y_min <- min(plot_data$GC_Ratio, na.rm = TRUE)
  y_range <- y_max - y_min
  p_value_y <- y_max + 0.25 * y_range
  plot_top <- y_max + 0.35 * y_range
  
  ggplot(plot_data, aes(x = Group, y = GC_Ratio, fill = Group)) +
    geom_violin(alpha = 0.7, width = 0.8) +
    geom_boxplot(width = 0.15, fill = "white", outlier.shape = NA) +
    
    # Wilcoxon test only between High vs Low ΔPSI
    stat_compare_means(
      comparisons = list(c("Low ΔPSI", "High ΔPSI")),
      method = "wilcox.test",
      label = "p = {p.format}",
      label.y = p_value_y,
      size = 3.5
    ) +
    
    scale_fill_manual(values = c(
      "Low ΔPSI" = "#21908C",
      "No ΔPSI" = "grey70",
      "High ΔPSI" = "#440154"
    )) +
    scale_y_continuous(limits = c(y_min, plot_top), expand = expansion(mult = c(0, 0.1))) +
    labs(title = title, subtitle = subtitle, y = "GC Ratio") +
    theme_minimal() +
    theme(
      legend.position = "none",
      plot.title = element_text(size = 10, face = "bold", hjust = 0.5),
      plot.subtitle = element_text(size = 8, hjust = 0.5)
    )
}

# -----------------------------
# 4️⃣ Dataset names, titles, labels
# -----------------------------
labels <- c(
  "Mattieu BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
  "Purcell MDAH041", "Zhang BJ", "Zhang BJ",
  "Zhang BJ", "Zhang BJ"
)
new_titles <- c(
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Pre-Senescent vs Proliferative",
  "Quiescent vs Proliferative",
  "Senescent vs Quiescent",
  "Deep-Senescent vs Proliferative",
  "Deep-Senescent vs Senescent"
)

# -----------------------------
# 5️⃣ Apply to all datasets
# -----------------------------
data_list_categorized <- lapply(data_list, add_psi_category)

violin_plots <- mapply(function(df, ttl, sub) {
  create_gc_plot(df, ttl, sub)
}, data_list_categorized, new_titles, labels, SIMPLIFY = FALSE)

# -----------------------------
# 6️⃣ Combine plots
# -----------------------------
final_plot <- wrap_plots(violin_plots, ncol = 4) +
  plot_annotation(title = "GC Content Comparison by ΔPSI",
                  theme = theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5)))

print(final_plot)






```



```{r}
library(patchwork)
library(ggplot2)

# --- 1️⃣ Create top 9 GC plots with subtitles ---
plots_top <- mapply(function(df, lbl) {
  create_gc_plot(df, title = "", subtitle = lbl)  # subtitle added
}, data_list_categorized[1:9], labels[1:9], SIMPLIFY = FALSE)

# --- 2️⃣ Fake title as ggplot for top group ---
top_title <- ggplot() +
  annotate("text", x = 0.5, y = 0.5, label = "Senescence vs Proliferative", size = 6, fontface = "bold") +
  theme_void()

# --- 3️⃣ Create bottom 5 GC plots with subtitles ---
plots_bottom <- mapply(function(df, lbl, ttl) {
  create_gc_plot(df, title = ttl, subtitle = lbl)
}, data_list_categorized[10:14], labels[10:14], new_titles[10:14], SIMPLIFY = FALSE)

# --- 4️⃣ Fill bottom to make 3x3 like top ---
num_spacers <- 9 - length(plots_bottom)
plots_bottom <- c(plots_bottom, rep(list(plot_spacer()), num_spacers))

# --- 5️⃣ Wrap grids ---
top_grid <- wrap_plots(plots_top, ncol = 3)
bottom_grid <- wrap_plots(plots_bottom, ncol = 3)

# --- 6️⃣ Dashed separator ---
separator <- ggplot() + 
  geom_hline(yintercept = 0, linetype = "dashed", color = "grey50") + 
  theme_void()

# --- 7️⃣ Stack everything ---
final_gc_plot <- top_title / top_grid / separator / bottom_grid +
  plot_layout(heights = c(0.2, 3, 0.05, 2), guides = "collect") &
  theme(legend.position = 'bottom')

# --- 8️⃣ Print final plot ---
print(final_gc_plot)




ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "gc_with_all_tryyyy.png"), 
  
  # 2. Specify which plot object to save
  plot = final_gc_plot,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 16,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```
```{r}
library(dplyr)
library(ggplot2)
library(patchwork)
library(ggpubr)

# -----------------------------
# 1️⃣ Define PSI categorization
# -----------------------------
add_psi_category <- function(df) {
  df %>%
    mutate(
      psi_category = case_when(
        PSI_difference > 5  ~ "PSI > 5",
        PSI_difference < -5 ~ "PSI < -5",
        between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
        TRUE ~ "Other"
      ),
      psi_category = factor(
        psi_category,
        levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other")
      )
    )
}

# -----------------------------
# 2️⃣ Prepare GC content data
# -----------------------------
prepare_gc_data <- function(df) {
  df %>%
    add_psi_category() %>%
    filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05")) %>%
    mutate(
      Group = case_when(
        psi_category == "PSI > 5"  ~ "High ΔPSI",
        psi_category == "PSI < -5" ~ "Low ΔPSI",
        psi_category == "-0.05 < PSI < 0.05" ~ "No ΔPSI"
      ),
      Group = factor(Group, levels = c("Low ΔPSI", "No ΔPSI", "High ΔPSI"))
    )
}

# -----------------------------
# 3️⃣ Create violin + boxplot
# -----------------------------
create_gc_plot <- function(df, title, subtitle) {
  plot_data <- prepare_gc_data(df) %>% filter(!is.na(GC_Ratio))
  
  if (nrow(plot_data) == 0) {
    return(ggplot() + labs(title = title, subtitle = "No data") + theme_void())
  }
  
  y_max <- max(plot_data$GC_Ratio, na.rm = TRUE)
  y_min <- min(plot_data$GC_Ratio, na.rm = TRUE)
  y_range <- y_max - y_min
  
  # Set different y positions for each comparison
  y_positions <- c(
    y_max + 0.35 * y_range,  # High vs Low
    y_max + 0.15 * y_range,  # High vs No ΔPSI
    y_max + 0.15 * y_range   # Low vs No ΔPSI
  )
  
  comparisons <- list(
    c("Low ΔPSI", "High ΔPSI"),
    c("No ΔPSI", "High ΔPSI"),
    c("Low ΔPSI", "No ΔPSI")
  )
  
  ggplot(plot_data, aes(x = Group, y = GC_Ratio, fill = Group)) +
    geom_violin(alpha = 0.7, width = 0.8) +
    geom_boxplot(width = 0.15, fill = "white", outlier.shape = NA) +
    
    # Add multiple pairwise comparisons
    stat_compare_means(
      comparisons = comparisons,
      method = "wilcox.test",
      label = "p = {p.format}", # <-- ADD THIS LINE to match your data frame
      label.y = y_positions,
      size = 3.5
    ) +
    scale_fill_manual(values = c(
      "Low ΔPSI" = "#21908C",
      "No ΔPSI" = "grey70",
      "High ΔPSI" = "#440154"
    )) +
    scale_y_continuous(limits = c(y_min, y_max + 0.5 * y_range), expand = expansion(mult = c(0, 0.1))) +
    labs(title = title, subtitle = subtitle, y = "GC Ratio") +
    theme_minimal() +
    theme(
      legend.position = "none",
      plot.title = element_text(size = 10, face = "bold", hjust = 0.5),
      plot.subtitle = element_text(size = 8, hjust = 0.5)
    )
}


# -----------------------------
# 4️⃣ Dataset names, titles, labels
# -----------------------------
labels <- c(
  "Deschênes BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
  "Purcell MDAH041", "Zhang BJ", "Zhang BJ", "Zhang BJ", "Zhang BJ"
)
new_titles <- c(
  "Senescent vs Proliferative", "Senescent vs Proliferative", "Senescent vs Proliferative",
  "Senescent vs Proliferative", "Senescent vs Proliferative", "Senescent vs Proliferative",
  "Senescent vs Proliferative", "Senescent vs Proliferative", "Senescent vs Proliferative",
  "Pre-Senescent vs Proliferative", "Quiescent vs Proliferative", "Senescent vs Quiescent",
  "Deep-Senescent vs Proliferative", "Deep-Senescent vs Senescent"
)

# -----------------------------
# 5️⃣ Apply to all datasets
# -----------------------------
data_list_categorized <- lapply(data_list, add_psi_category)

violin_plots <- mapply(function(df, ttl, sub) {
  create_gc_plot(df, ttl, sub)
}, data_list_categorized, new_titles, labels, SIMPLIFY = FALSE)

# -----------------------------
# 6️⃣ Combine plots
# -----------------------------
final_plot <- wrap_plots(violin_plots, ncol = 4) +
  plot_annotation(
    title = "GC Content Comparison by ΔPSI",
    theme = theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5))
  )

# --- Patchwork layout with subtitles and dashed separator ---
# --- 1️⃣ Top 9 plots ---
plots_top <- mapply(function(df, lbl) {
  create_gc_plot(df, title = "", subtitle = lbl)
}, data_list_categorized[1:9], labels[1:9], SIMPLIFY = FALSE)

# --- 2️⃣ Fake title for top group ---
top_title <- ggplot() +
  annotate("text", x = 0.5, y = 0.5, label = "Senescence vs Proliferative", size = 6, fontface = "bold") +
  theme_void()

# --- 3️⃣ Bottom 5 plots ---
plots_bottom <- mapply(function(df, lbl, ttl) {
  create_gc_plot(df, title = ttl, subtitle = lbl)
}, data_list_categorized[10:14], labels[10:14], new_titles[10:14], SIMPLIFY = FALSE)

# --- 4️⃣ Fill bottom to 3x3 grid ---
num_spacers <- 9 - length(plots_bottom)
plots_bottom <- c(plots_bottom, rep(list(plot_spacer()), num_spacers))

# --- 5️⃣ Wrap grids ---
top_grid <- wrap_plots(plots_top, ncol = 3)
bottom_grid <- wrap_plots(plots_bottom, ncol = 3)

# --- 6️⃣ Dashed separator ---
separator <- ggplot() +
  geom_hline(yintercept = 0, linetype = "dashed", color = "grey50") +
  theme_void()

# --- 7️⃣ Stack everything ---
final_gc_plot <- top_title / top_grid / separator / bottom_grid +
  plot_layout(heights = c(0.2, 3, 0.05, 2), guides = "collect") &
  theme(legend.position = 'bottom')

# --- 8️⃣ Print final plot ---
print(final_gc_plot)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "gc_with_all_tryyyy.png"), 
  
  # 2. Specify which plot object to save
  plot = final_gc_plot,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 16,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)


```


### STATISTICAL FOR GC


```{r}
library(dplyr)
library(tidyr)

data_list <- list(
  data1, data4, data6, data8, data9, data10, data11,
  data12, data14, data13, data3, data7, data2, data5
)

names(data_list) <- c(
  "data1", "data4", "data6", "data8", "data9", "data10", "data11",
  "data12", "data14", "data13", "data3", "data7", "data2", "data5"
)

# --- Function to run stats and extract results in tidy format ---
extract_stats <- function(df, dataset_name) {
  # Prepare psi_group
  plot_data_gc <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference <= 0.05 & PSI_difference >= -0.05 ~ "Near Zero (|ΔPSI| < 0.05)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(GC_Ratio)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.05)")
    ))
  
  # Kruskal-Wallis test (no log)
  kruskal_res <- kruskal.test(GC_Ratio ~ psi_group, data = plot_data_gc)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_gc$GC_Ratio,
    plot_data_gc$psi_group,
    p.adjust.method = "none"
  )
  
  # Tidy pairwise results
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # --- Compute medians for each psi_group ---
  medians <- plot_data_gc %>%
    group_by(psi_group) %>%
    summarise(median_gc = median(GC_Ratio, na.rm = TRUE), .groups = "drop")
  
  # Add dataset, Kruskal-Wallis info, and group medians
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_gc) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_gc) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Loop over all datasets ---
all_stats_gc <- mapply(
  extract_stats, 
  df = data_list, 
  dataset_name = names(data_list), 
  SIMPLIFY = FALSE
)

# Combine into a single data frame
all_stats_gc_df <- bind_rows(all_stats_gc)


all_stats_gc_df

# --- Optional: write CSV ---
write.csv(all_stats_gc_df, "C:\\Users\\35196\\Desktop\\all_datasets_stat_tests_GC_Ratio.csv", row.names = FALSE)

# --- Optional: view first few rows ---
head(all_stats_gc_df)

```







# 3 ' splice site 


```{r}
library(dplyr)
library(ggplot2)
library(patchwork)
library(ggpubr)

# -----------------------------
# 1️⃣ Define PSI categorization
# -----------------------------
add_psi_category <- function(df) {
  df %>%
    mutate(
      psi_category = case_when(
        PSI_difference > 5  ~ "PSI > 5",
        PSI_difference < -5 ~ "PSI < -5",
        between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
        TRUE ~ "Other"
      ),
      psi_category = factor(
        psi_category,
        levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other")
      )
    )
}

# -----------------------------
# 2️⃣ Prepare 3_score data
# -----------------------------
prepare_3score_data <- function(df) {
  df %>%
    add_psi_category() %>%
    filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05")) %>%
    mutate(Group = case_when(
      psi_category == "PSI > 5"  ~ "High ΔPSI",
      psi_category == "PSI < -5" ~ "Low ΔPSI",
      psi_category == "-0.05 < PSI < 0.05" ~ "No ΔPSI"
    ),
    Group = factor(Group, levels = c("Low ΔPSI", "No ΔPSI", "High ΔPSI")))
}

# -----------------------------
# 3️⃣ Create violin + boxplot
# -----------------------------
create_3score_plot <- function(df, title, subtitle) {
  plot_data <- prepare_3score_data(df) %>% filter(!is.na(`3_score`))
  
  if(nrow(plot_data) == 0){
    return(ggplot() + labs(title = title, subtitle = "No data") + theme_void())
  }
  
  y_max <- max(plot_data$`3_score`, na.rm = TRUE)
  y_min <- min(plot_data$`3_score`, na.rm = TRUE)
  y_range <- y_max - y_min
  
  # Define multiple y positions for closer spacing
  y_positions <- c(
    y_max + 0.15 * y_range,  # High vs Low
    y_max + 0.05 * y_range,  # High vs No ΔPSI
    y_max + 0.05 * y_range   # Low vs No ΔPSI
  )
  
  # All pairwise comparisons
  comparisons <- list(
    c("Low ΔPSI", "High ΔPSI"),
    c("No ΔPSI", "High ΔPSI"),
    c("Low ΔPSI", "No ΔPSI")
  )
  
  ggplot(plot_data, aes(x = Group, y = `3_score`, fill = Group)) +
    geom_violin(alpha = 0.7, width = 0.8) +
    geom_boxplot(width = 0.15, fill = "white", outlier.shape = NA) +
    
    # Wilcoxon tests for all comparisons
    stat_compare_means(
      comparisons = comparisons,
      method = "wilcox.test",
      label = "p = {p.format}",
      label.y = y_positions,
      size = 3.5
    ) +
    
    scale_fill_manual(values = c(
      "Low ΔPSI" = "#21908C",
      "No ΔPSI" = "grey70",
      "High ΔPSI" = "#440154"
    )) +
    scale_y_continuous(limits = c(y_min, max(y_positions) + 0.05 * y_range), expand = expansion(mult = c(0, 0.05))) +
    labs(title = title, subtitle = subtitle, y = "3_score") +
    theme_minimal() +
    theme(
      legend.position = "none",
      plot.title = element_text(size = 10, face = "bold", hjust = 0.5),
      plot.subtitle = element_text(size = 8, hjust = 0.5)
    )
}



# -----------------------------
# 4️⃣ Dataset names, titles, labels
# -----------------------------
labels <- c(
  "Deschênes BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
  "Purcell MDAH041", "Zhang BJ", "Zhang BJ",
  "Zhang BJ", "Zhang BJ"
)

new_titles <- c(
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Senescent vs Proliferative",
  "Pre-Senescent vs Proliferative",
  "Quiescent vs Proliferative",
  "Senescent vs Quiescent",
  "Deep-Senescent vs Proliferative",
  "Deep-Senescent vs Senescent"
)

# -----------------------------
# 5️⃣ Apply to all datasets
# -----------------------------
data_list_categorized <- lapply(data_list, add_psi_category)

violin_plots_3score <- mapply(function(df, ttl, sub) {
  create_3score_plot(df, ttl, sub)
}, data_list_categorized, new_titles, labels, SIMPLIFY = FALSE)

# -----------------------------
# 6️⃣ Combine plots
# -----------------------------
final_plot_3score <- wrap_plots(violin_plots_3score, ncol = 4) +
  plot_annotation(title = "3_score Comparison by ΔPSI",
                  theme = theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5)))

print(final_plot_3score)





```

### STATISTICAL FOR 3



```{r}
library(dplyr)
library(tidyr)

data_list <- list(
  data1, data4, data6, data8, data9, data10, data11,
  data12, data14, data13, data3, data7, data2, data5
)

names(data_list) <- c(
  "data1", "data4", "data6", "data8", "data9", "data10", "data11",
  "data12", "data14", "data13", "data3", "data7", "data2", "data5"
)

# --- Function to run stats and extract results in tidy format ---
extract_stats <- function(df, dataset_name) {
  # Prepare psi_group
  plot_data_score <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference <= 0.05 & PSI_difference >= -0.05 ~ "Near Zero (|ΔPSI| < 0.05)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(`3_score`)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.05)")
    ))
  
  # Kruskal-Wallis test (no log)
  kruskal_res <- kruskal.test(`3_score` ~ psi_group, data = plot_data_score)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_score$`3_score`,
    plot_data_score$psi_group,
    p.adjust.method = "none")
  
  # Tidy pairwise results
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # --- Compute medians for each psi_group ---
  medians <- plot_data_score %>%
    group_by(psi_group) %>%
    summarise(median_score = median(`3_score`, na.rm = TRUE), .groups = "drop")
  
  # Add dataset, Kruskal-Wallis info, and group medians
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_score) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_score) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Loop over all datasets ---
all_stats_3score <- mapply(
  extract_stats, 
  df = data_list, 
  dataset_name = names(data_list), 
  SIMPLIFY = FALSE
)

# Combine into a single data frame
all_stats_3score_df <- bind_rows(all_stats_3score)

# --- Optional: write CSV ---
write.csv(all_stats_3score_df, "C:\\Users\\35196\\Desktop\\all_datasets_stat_tests_3_score.csv", row.names = FALSE)

# --- Optional: view first few rows ---
head(all_stats_3score_df)

```



```{r}
library(patchwork)
library(ggplot2)

# --- 1️⃣ Top 9 3_score plots ---
plots_top <- mapply(function(df, lbl) {
  create_3score_plot(df, title = "", subtitle = lbl)  # remove individual titles
}, data_list_categorized[1:9], labels[1:9], SIMPLIFY = FALSE)

# --- 2️⃣ Fake title as ggplot object ---
top_title <- ggplot() +
  annotate("text", x = 0.5, y = 0.5, label = "Senescence vs Proliferative", size = 6, fontface = "bold") +
  theme_void()

# --- 3️⃣ Bottom 5 3_score plots ---
plots_bottom <- mapply(function(df, lbl, ttl) {
  create_3score_plot(df, title = ttl, subtitle = lbl)
}, data_list_categorized[10:14], labels[10:14], new_titles[10:14], SIMPLIFY = FALSE)

# --- 4️⃣ Fill bottom to make 3x3 grid for alignment ---
num_spacers <- 9 - length(plots_bottom)
plots_bottom <- c(plots_bottom, rep(list(plot_spacer()), num_spacers))

# --- 5️⃣ Wrap grids ---
top_grid <- wrap_plots(plots_top, ncol = 3)
bottom_grid <- wrap_plots(plots_bottom, ncol = 3)

# --- 6️⃣ Dashed separator ---
separator <- ggplot() + geom_hline(yintercept = 0, linetype = "dashed", color = "grey50") + theme_void()

# --- 7️⃣ Stack everything ---
final_plot_3score <- top_title / top_grid / separator / bottom_grid +
  plot_layout(heights = c(0.2, 3, 0.05, 2), guides = "collect") &
  theme(legend.position = 'bottom')

# --- 8️⃣ Print final plot ---
print(final_plot_3score)




ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "3score_plot_all_tryyy.png"), 
  
  # 2. Specify which plot object to save
  plot = final_plot_3score,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 18,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)



```




5555555555555



```{r}
  library(dplyr)
  library(ggplot2)
  library(patchwork)
  library(ggpubr)
  
  # -----------------------------
  # 1️⃣ Define PSI categorization
  # -----------------------------
  add_psi_category <- function(df) {
    df %>%
      mutate(
        psi_category = case_when(
          PSI_difference > 5  ~ "PSI > 5",
          PSI_difference < -5 ~ "PSI < -5",
          between(PSI_difference, -0.05, 0.05) ~ "-0.05 < PSI < 0.05",
          TRUE ~ "Other"
        ),
        psi_category = factor(
          psi_category,
          levels = c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05", "Other")
        )
      )
  }
  
  # -----------------------------
  # 2️⃣ Prepare 5_score data
  # -----------------------------
  prepare_5score_data <- function(df) {
    df %>%
      add_psi_category() %>%
      filter(psi_category %in% c("PSI > 5", "PSI < -5", "-0.05 < PSI < 0.05")) %>%
      mutate(Group = case_when(
        psi_category == "PSI > 5"  ~ "High ΔPSI",
        psi_category == "PSI < -5" ~ "Low ΔPSI",
        psi_category == "-0.05 < PSI < 0.05" ~ "No ΔPSI"
      ),
      Group = factor(Group, levels = c("Low ΔPSI", "No ΔPSI", "High ΔPSI")))
  }
  
  # -----------------------------
  # 3️⃣ Create violin + boxplot
  # -----------------------------
  create_5score_plot <- function(df, title, subtitle) {
  plot_data <- prepare_5score_data(df) %>% filter(!is.na(`5_score`))
  
  if(nrow(plot_data) == 0){
    return(ggplot() + labs(title = title, subtitle = "No data") + theme_void())
  }
  
  comparisons <- list(
    c("Low ΔPSI", "High ΔPSI"),
    c("No ΔPSI", "High ΔPSI"),
    c("Low ΔPSI", "No ΔPSI")
  )
  
  # Calculate y_max and y_range
  y_max <- max(plot_data$`5_score`, na.rm = TRUE)
  y_min <- min(plot_data$`5_score`, na.rm = TRUE)
  y_range <- y_max - y_min
  
  # Set separate y positions for each comparison
  y_positions <- c(
    y_max + 0.15 * y_range,  # Low vs High
    y_max + 0.05 * y_range,  # No vs High
    y_max + 0.05 * y_range   # Low vs No
  )
  
  # Make sure the highest y_position does not exceed 20
  y_positions <- pmin(y_positions, 20)
  
  ggplot(plot_data, aes(x = Group, y = `5_score`, fill = Group)) +
    geom_violin(alpha = 0.7, width = 0.8) +
    geom_boxplot(width = 0.15, fill = "white", outlier.shape = NA) +
    
    stat_compare_means(
      comparisons = comparisons,
      method = "wilcox.test",
      label = "p = {p.format}",
      label.y = y_positions,
      size = 3.5
    ) +
    
    scale_fill_manual(values = c(
      "Low ΔPSI" = "#21908C",
      "No ΔPSI" = "grey70",
      "High ΔPSI" = "#440154"
    )) +
    
    # Limit the y-axis max at 20
    scale_y_continuous(limits = c(y_min, 20), expand = expansion(mult = c(0, 0.05))) +
    
    labs(title = title, subtitle = subtitle, y = "5_score") +
    theme_minimal() +
    theme(
      legend.position = "none",
      plot.title = element_text(size = 10, face = "bold", hjust = 0.5),
      plot.subtitle = element_text(size = 8, hjust = 0.5)
    )
}




  
  # -----------------------------
  # 4️⃣ Dataset names, titles, labels
  # -----------------------------
  labels <- c(
    "Deschênes BJ", "Zhang BJ", "Casella IMR-90", "Marthandan BJ", "Marthandan HFF",
    "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041", "Suda WI-38",
    "Purcell MDAH041", "Zhang BJ", "Zhang BJ",
    "Zhang BJ", "Zhang BJ"
  )
  
  new_titles <- c(
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Senescent vs Proliferative",
    "Pre-Senescent vs Proliferative",
    "Quiescent vs Proliferative",
    "Senescent vs Quiescent",
    "Deep-Senescent vs Proliferative",
    "Deep-Senescent vs Senescent"
  )
  
  # -----------------------------
  # 5️⃣ Apply to all datasets
  # -----------------------------
  data_list_categorized <- lapply(data_list, add_psi_category)
  
  violin_plots_5score <- mapply(function(df, ttl, sub) {
    create_5score_plot(df, ttl, sub)
  }, data_list_categorized, new_titles, labels, SIMPLIFY = FALSE)
  
  # -----------------------------
  # 6️⃣ Combine plots
  # -----------------------------
  final_plot_5score <- wrap_plots(violin_plots_5score, ncol = 4) +
    plot_annotation(title = "5_score Comparison by ΔPSI",
                    theme = theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5)))
  
  print(final_plot_5score)





```




```{r}
library(patchwork)
library(ggplot2)

# --- 1️⃣ Top 9 5_score plots ---
plots_top <- mapply(function(df, lbl) {
  create_5score_plot(df, title = "", subtitle = lbl)  # remove individual titles
}, data_list_categorized[1:9], labels[1:9], SIMPLIFY = FALSE)

# --- 2️⃣ Fake title as ggplot object ---
top_title <- ggplot() +
  annotate("text", x = 0.5, y = 0.5, label = "Senescence vs Proliferative", size = 6, fontface = "bold") +
  theme_void()

# --- 3️⃣ Bottom 5 5_score plots ---
plots_bottom <- mapply(function(df, lbl, ttl) {
  create_5score_plot(df, title = ttl, subtitle = lbl)
}, data_list_categorized[10:14], labels[10:14], new_titles[10:14], SIMPLIFY = FALSE)

# --- 4️⃣ Fill bottom to make 3x3 grid for alignment ---
num_spacers <- 9 - length(plots_bottom)
plots_bottom <- c(plots_bottom, rep(list(plot_spacer()), num_spacers))

# --- 5️⃣ Wrap grids ---
top_grid <- wrap_plots(plots_top, ncol = 3)
bottom_grid <- wrap_plots(plots_bottom, ncol = 3)

# --- 6️⃣ Dashed separator ---
separator <- ggplot() + geom_hline(yintercept = 0, linetype = "dashed", color = "grey50") + theme_void()

# --- 7️⃣ Stack everything ---
final_plot_5score <- top_title / top_grid / separator / bottom_grid +
  plot_layout(heights = c(0.2, 3, 0.05, 2), guides = "collect") &
  theme(legend.position = 'bottom')

# --- 8️⃣ Print final plot ---
print(final_plot_5score)



ggsave(
  # 1. Define the full file path and name
  filename = file.path("C:\\Users\\35196\\Desktop\\Plots", "5score_plot_all_tryyy.png"), 
  
  # 2. Specify which plot object to save
  plot = final_plot_5score,
  
  # 3. Set the dimensions and resolution
  width = 12,
  height = 18,
  units = "in",
  dpi = 300,      # High resolution for print quality
  
  # 4. This is the key: specify the Cairo backend for a beautiful output
  type = "cairo"
)

```

### STATISTICAL FOR 5



```{r}
library(dplyr)
library(tidyr)


data_list <- list(
  data1, data4, data6, data8, data9, data10, data11,
  data12, data14, data13, data3, data7, data2, data5
)

names(data_list) <- c(
  "data1", "data4", "data6", "data8", "data9", "data10", "data11",
  "data12", "data14", "data13", "data3", "data7", "data2", "data5"
)

# --- Function to run stats and extract results in tidy format ---
extract_stats <- function(df, dataset_name) {
  # Prepare psi_group
  plot_data_score <- df %>%
    mutate(psi_group = case_when(
      PSI_difference > 5                             ~ "High ΔPSI (> 5)",
      PSI_difference < -5                            ~ "Low ΔPSI (< -5)",
      PSI_difference <= 0.05 & PSI_difference >= -0.05 ~ "Near Zero (|ΔPSI| < 0.05)",
      TRUE                                           ~ NA_character_
    )) %>%
    filter(!is.na(psi_group), !is.na(`5_score`)) %>%
    mutate(psi_group = factor(
      psi_group,
      levels = c("High ΔPSI (> 5)", "Low ΔPSI (< -5)", "Near Zero (|ΔPSI| < 0.05)")
    ))
  
  # Kruskal-Wallis test (no log)
  kruskal_res <- kruskal.test(`5_score` ~ psi_group, data = plot_data_score)
  
  # Pairwise Wilcoxon test
  pairwise_res <- pairwise.wilcox.test(
    plot_data_score$`5_score`,
    plot_data_score$psi_group,
    p.adjust.method = "none"
  )
  
  # Tidy pairwise results
  pairwise_df <- as.data.frame(pairwise_res$p.value) %>%
    tibble::rownames_to_column(var = "group1") %>%
    pivot_longer(-group1, names_to = "group2", values_to = "p_value") %>%
    filter(!is.na(p_value))
  
  # --- Compute medians for each psi_group ---
  medians <- plot_data_score %>%
    group_by(psi_group) %>%
    summarise(median_score = median(`5_score`, na.rm = TRUE), .groups = "drop")
  
  # Add dataset, Kruskal-Wallis info, and group medians
  pairwise_df <- pairwise_df %>%
    mutate(
      dataset = dataset_name,
      kruskal_statistic = kruskal_res$statistic,
      kruskal_p_value = kruskal_res$p.value
    ) %>%
    left_join(medians, by = c("group1" = "psi_group")) %>%
    rename(median_group1 = median_score) %>%
    left_join(medians, by = c("group2" = "psi_group")) %>%
    rename(median_group2 = median_score) %>%
    select(dataset, kruskal_statistic, kruskal_p_value,
           group1, group2, p_value, median_group1, median_group2)
  
  return(pairwise_df)
}

# --- Loop over all datasets ---
all_stats_5score <- mapply(
  extract_stats, 
  df = data_list, 
  dataset_name = names(data_list), 
  SIMPLIFY = FALSE
)

# Combine into a single data frame
all_stats_5score_df <- bind_rows(all_stats_5score)

# --- Optional: write CSV ---
write.csv(all_stats_5score_df, "C:\\Users\\35196\\Desktop\\all_datasets_stat_tests_5_score.csv", row.names = FALSE)

# --- Optional: view first few rows ---
head(all_stats_5score_df)

```



# GO


```{r}
library(clusterProfiler)
library(org.Hs.eg.db)
library(dplyr)
library(ggplot2)
library(patchwork)

# Your data list
data_list_high <- list(
  data1_high, data2_high, data3_high, data4_high, data5_high, data6_high, 
  data7_high, data8_high, data9_high, data10_high, data11_high, data12_high, 
  data13_high, data14_high
)
names(data_list_high) <- paste0("data", 1:14)

# Dataset labels for subtitles
labels <- c(
  "Mattieu BJ", "Zhang BJ Deep", "Zhang BJ Quiescent", "Zhang BJ Senescence", "Zhang BJ Senescence vs Deep",
  "Zhang BJ Quiescent vs Senescence", "Casella IMR-90 Ribodepletion Senescence", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041 Senescence", "Purcell MDAH041 Pre", "Suda WI-38 Senescence"
)

# Improved function to extract and validate gene symbols
get_valid_genes <- function(df) {
  # Ensure the data frame has a "Gene" column
  if (!"Gene" %in% colnames(df)) {
    warning("Data frame doesn't contain 'Gene' column")
    return(character(0))
  }
  
  # Get unique, non-NA genes
  genes <- unique(na.omit(df$Gene))
  
  # Remove genes that are empty strings
  genes <- genes[genes != ""]
  
  if (length(genes) == 0) {
    warning("No valid genes found in dataset")
    return(character(0))
  }
  
  # Check which genes are mappable
  mapped_genes <- suppressMessages(
    bitr(genes, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Hs.eg.db)
  )
  
  if (nrow(mapped_genes) == 0) {
    warning("No genes could be mapped to ENTREZID")
    return(character(0))
  }
  
  return(mapped_genes$SYMBOL)
}

# Run GO enrichment with better error handling
run_go_enrichment <- function(genes) {
  if (length(genes) < 5) {
    message("Not enough genes for enrichment analysis (need at least 5)")
    return(NULL)
  }
  
  tryCatch({
    enrichGO(
      gene = genes,
      OrgDb = org.Hs.eg.db,
      keyType = "SYMBOL",
      ont = "BP",
      pvalueCutoff = 0.05,
      pAdjustMethod = "BH",
      readable = TRUE,
      minGSSize = 10,
      maxGSSize = 500
    )
  }, error = function(e) {
    message("Error in GO enrichment: ", e$message)
    return(NULL)
  })
}

# Robust plotting function with alternative implementation
plot_go_bar <- function(go_result, dataset_name, label_text) {
  if (is.null(go_result) || nrow(go_result@result) == 0) {
    message("No significant GO terms for ", dataset_name)
    # Return empty plot with message
    return(
      ggplot() + 
        annotate("text", x = 1, y = 1, size = 5,
                 label = "No significant GO terms") +
        labs(title = paste("GO BP -", dataset_name),
             subtitle = label_text) +
        theme_void()
    )
  }
  
  # Prepare data - handle cases where there are fewer than 10 terms
  plot_data <- go_result@result
  if (nrow(plot_data) > 10) {
    plot_data <- plot_data[1:10, ]
  }
  
  # Shorten long descriptions
  plot_data$ShortDescription <- substr(plot_data$Description, 1, 50)
  
  # Create factor levels to maintain order
  plot_data$ShortDescription <- factor(plot_data$ShortDescription,
                                     levels = rev(plot_data$ShortDescription))
  
  ggplot(plot_data, aes(x = -log10(pvalue), y = ShortDescription)) +
    geom_col(fill = "steelblue", width = 0.8) +
    labs(title = paste("GO BP -", dataset_name),
         subtitle = label_text,
         x = "-log10(p-value)",
         y = "") +
    theme_minimal() +
    theme(
      plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
      plot.subtitle = element_text(size = 10, hjust = 0.5, face = "italic"),
      axis.text.y = element_text(size = 8),
      panel.grid.major.y = element_blank(),
      panel.grid.minor.y = element_blank()
    )
}

# Run analysis with comprehensive error handling
go_plots_high <- list()
for (i in seq_along(data_list_high)) {
  dataset_name <- names(data_list_high)[i]
  label_text <- labels[i]
  df <- data_list_high[[i]]
  
  message("\n=== Processing ", dataset_name, " (", label_text, ") ===")
  
  tryCatch({
    # Step 1: Get valid genes
    genes <- get_valid_genes(df)
    if (length(genes) == 0) {
      message("-> No valid genes, skipping")
      next
    }
    message("-> Found ", length(genes), " valid genes")
    if (length(genes) > 5) {
      message("-> Gene examples: ", paste(sample(genes, 5), collapse = ", "), ", ...")
    } else {
      message("-> Genes: ", paste(genes, collapse = ", "))
    }
    
    # Step 2: Run GO enrichment
    go_res <- NULL
    tryCatch({
      go_res <- run_go_enrichment(genes)
    }, error = function(e) {
      message("-> GO enrichment error: ", e$message)
    })
    
    if (is.null(go_res)) {
      message("-> GO enrichment failed")
      next
    }
    message("-> Found ", nrow(go_res@result), " significant GO terms")
    if (nrow(go_res@result) > 0) {
      message("-> Top term: ", go_res@result$Description[1], 
             " (p=", signif(go_res@result$pvalue[1], 3), ")")
    }
    
    # Step 3: Create plot
    p <- NULL
    tryCatch({
      p <- plot_go_bar(go_res, dataset_name, label_text)
    }, error = function(e) {
      message("-> Plot creation failed: ", e$message)
    })
    
    if (!is.null(p)) {
      go_plots_high[[dataset_name]] <- p
      message("-> Successfully created plot")
    } else {
      message("-> No plot created")
    }
    
  }, error = function(e) {
    message("!! Critical error processing dataset ", dataset_name, ": ", e$message)
  })
}

# Plotting the results with multiple fallback options
if (length(go_plots_high) > 0) {
  # First try patchwork combination
  tryCatch({
    n_plots <- length(go_plots_high)
    n_col <- min(3, ceiling(sqrt(n_plots)))
    
    combined_plot <- wrap_plots(go_plots_high, ncol = n_col) +
      plot_annotation(
        title = "GO Biological Processes for ΔPSI > 5 Events",
        subtitle = paste("Showing results for", length(go_plots_high), "of", length(data_list_high), "datasets"),
        theme = theme(
          plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
          plot.subtitle = element_text(size = 12, hjust = 0.5)
        )
      )
    
    print(combined_plot)
  }, error = function(e) {
    message("Could not combine plots with patchwork: ", e$message)
    message("Falling back to individual plots...")
    
    # Plot each one individually
    for (i in seq_along(go_plots_high)) {
      plot_title <- paste("Dataset:", names(go_plots_high)[i], 
                         "-", labels[which(names(data_list_high) == names(go_plots_high)[i])])
      
      print(go_plots_high[[i]] + 
              ggtitle(plot_title) +
              theme(plot.title = element_text(size = 12, face = "bold")))
      
      # Add pause if viewing interactively
      if (interactive() && i < length(go_plots_high)) {
        readline(prompt = "Press [Enter] to see next plot...")
      }
    }
  })
} else {
  message("No valid GO results to plot from any dataset")
}

# Special debugging output for dataset 4 if it failed
if ("data4" %in% names(data_list_high) && (!"data4" %in% names(go_plots_high))) {
  message("\n=== Debugging output for dataset 4 (Zhang BJ Senescence) ===")
  
  data4 <- data_list_high[["data4"]]
  
  # 1. Check data structure
  message("\nData structure:")
  print(str(data4, max.level = 1))
  
  # 2. Check gene column
  message("\nGene column summary:")
  if ("Gene" %in% colnames(data4)) {
    print(summary(data4$Gene))
    message("NA count: ", sum(is.na(data4$Gene)))
    message("Empty string count: ", sum(data4$Gene == "", na.rm = TRUE))
    message("Unique genes: ", length(unique(na.omit(data4$Gene))))
  } else {
    message("No 'Gene' column found")
  }
  
  # 3. Try gene mapping
  genes4 <- get_valid_genes(data4)
  message("\nGene mapping results (", length(genes4), " mappable genes):")
  if (length(genes4) > 0) {
    print(head(genes4))
  }
  
  # 4. Try GO enrichment
  if (length(genes4) >= 5) {
    message("\nAttempting GO enrichment...")
    go4 <- run_go_enrichment(genes4)
    if (!is.null(go4)) {
      message("GO enrichment successful")
      if (nrow(go4@result) > 0) {
        message("Top 5 terms:")
        print(head(go4@result[, c("Description", "pvalue", "Count")], 5))
      } else {
        message("No significant terms found")
      }
    }
  } else {
    message("Not enough mappable genes for GO enrichment (need ≥5, found ", length(genes4), ")")
  }
}



library(clusterProfiler)
library(org.Hs.eg.db)
library(ggplot2)

# Your data and labels
data_list_high <- list(
  data1_high, data2_high, data3_high, data4_high, data5_high, data6_high, 
  data7_high, data8_high, data9_high, data10_high, data11_high, data12_high, 
  data13_high, data14_high
)
names(data_list_high) <- paste0("data", 1:14)

labels <- c(
  "Mattieu BJ", "Zhang BJ Deep", "Zhang BJ Quiescent", "Zhang BJ Senescence", 
  "Zhang BJ Senescence vs Deep", "Zhang BJ Quiescent vs Senescence", 
  "Casella IMR-90 Ribodepletion Senescence", "Marthandan BJ", "Marthandan HFF",
  "Marthandan IMR90", "Marthandan WI-38", "Purcell MDAH041 Senescence", 
  "Purcell MDAH041 Pre", "Suda WI-38 Senescence"
)

# Create directory for plots
if (!dir.exists("GO_plots")) dir.create("GO_plots")

# Function to process and plot each dataset
plot_individual_go <- function(df, dataset_name, label) {
  # Get valid genes
  genes <- unique(na.omit(df$Gene))
  genes <- genes[genes != ""]
  
  if (length(genes) < 5) {
    message("Skipping ", dataset_name, " (", label, "): Not enough genes (", length(genes), ")")
    return(FALSE)
  }
  
  # Map genes
  gene_map <- tryCatch(
    bitr(genes, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Hs.eg.db),
    error = function(e) NULL
  )
  
  if (is.null(gene_map) || nrow(gene_map) == 0) {
    message("Skipping ", dataset_name, " (", label, "): Gene mapping failed")
    return(FALSE)
  }
  
  # Run GO analysis
  go_res <- tryCatch(
    enrichGO(
      gene = gene_map$SYMBOL,
      OrgDb = org.Hs.eg.db,
      keyType = "SYMBOL",
      ont = "BP",
      pvalueCutoff = 0.05,
      readable = TRUE
    ),
    error = function(e) NULL
  )
  
  if (is.null(go_res) || nrow(go_res@result) == 0) {
    message("Skipping ", dataset_name, " (", label, "): No GO results")
    return(FALSE)
  }
  
  # Prepare plot data
  plot_data <- head(go_res@result[order(go_res@result$pvalue), ], 10)
  plot_data$Description <- factor(
    substr(plot_data$Description, 1, 50),
    levels = rev(substr(plot_data$Description, 1, 50))
  )
  
  # Create plot
  p <- ggplot(plot_data, aes(x = -log10(pvalue), y = Description)) +
    geom_col(fill = "steelblue", width = 0.7) +
    labs(
      title = paste("GO Analysis:", label),
      subtitle = paste("Dataset:", dataset_name),
      x = "-log10(p-value)",
      y = ""
    ) +
    theme_minimal() +
    theme(
      plot.title = element_text(size = 12, face = "bold"),
      plot.subtitle = element_text(size = 10),
      axis.text.y = element_text(size = 8)
    )
  
  # Save to file
  filename <- paste0("GO_plots/", dataset_name, "_", gsub("[^[:alnum:]]", "_", label), ".png")
  ggsave(filename, plot = p, width = 8, height = 6, dpi = 300)
  
  return(TRUE)
}

# Process all datasets automatically
results <- mapply(function(df, name, label) {
  message("\nProcessing ", name, ": ", label)
  plot_individual_go(df, name, label)
}, data_list_high, names(data_list_high), labels)

# Summary of results
message("\nProcessing complete!")
message("Successful: ", sum(results))
message("Failed: ", sum(!results))
message("Plots saved in 'GO_plots' directory")
```


